<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ul.lst-kix_nh8twm5hcw9o-2{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-3{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-0{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-1{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-6{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-7{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-4{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-5{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-8{list-style-type:none}.lst-kix_vknp6nujnm9e-3>li:before{content:"\0025cf  "}.lst-kix_vknp6nujnm9e-5>li:before{content:"\0025a0  "}.lst-kix_vknp6nujnm9e-4>li:before{content:"\0025cb  "}.lst-kix_vknp6nujnm9e-7>li:before{content:"\0025cb  "}.lst-kix_vknp6nujnm9e-6>li:before{content:"\0025cf  "}.lst-kix_vknp6nujnm9e-8>li:before{content:"\0025a0  "}ul.lst-kix_vknp6nujnm9e-8{list-style-type:none}ul.lst-kix_vknp6nujnm9e-6{list-style-type:none}ul.lst-kix_vknp6nujnm9e-7{list-style-type:none}ul.lst-kix_vknp6nujnm9e-4{list-style-type:none}ul.lst-kix_vknp6nujnm9e-5{list-style-type:none}ul.lst-kix_vknp6nujnm9e-2{list-style-type:none}ul.lst-kix_vknp6nujnm9e-3{list-style-type:none}.lst-kix_wulsapsxo6ef-7>li:before{content:"\0025cb  "}.lst-kix_wulsapsxo6ef-8>li:before{content:"\0025a0  "}.lst-kix_wulsapsxo6ef-2>li:before{content:"\0025a0  "}.lst-kix_wulsapsxo6ef-1>li:before{content:"\0025cb  "}.lst-kix_wulsapsxo6ef-3>li:before{content:"\0025cf  "}ul.lst-kix_vknp6nujnm9e-0{list-style-type:none}.lst-kix_wulsapsxo6ef-6>li:before{content:"\0025cf  "}ul.lst-kix_vknp6nujnm9e-1{list-style-type:none}.lst-kix_wulsapsxo6ef-5>li:before{content:"\0025a0  "}.lst-kix_wulsapsxo6ef-4>li:before{content:"\0025cb  "}.lst-kix_nh8twm5hcw9o-4>li:before{content:"\0025cb  "}.lst-kix_weixxh5ko91p-0>li:before{content:"\0025cf  "}.lst-kix_cbfpvidehrpt-3>li:before{content:"\0025cf  "}.lst-kix_cbfpvidehrpt-7>li:before{content:"\0025cb  "}.lst-kix_nh8twm5hcw9o-0>li:before{content:"\0025cf  "}.lst-kix_nh8twm5hcw9o-8>li:before{content:"\0025a0  "}.lst-kix_weixxh5ko91p-4>li:before{content:"\0025cb  "}.lst-kix_d0pnlbhmephp-6>li:before{content:"\0025cf  "}.lst-kix_nh8twm5hcw9o-2>li:before{content:"\0025a0  "}.lst-kix_cbfpvidehrpt-1>li:before{content:"\0025cb  "}ul.lst-kix_hefi1acv3kbb-0{list-style-type:none}ul.lst-kix_hefi1acv3kbb-1{list-style-type:none}.lst-kix_weixxh5ko91p-6>li:before{content:"\0025cf  "}.lst-kix_weixxh5ko91p-8>li:before{content:"\0025a0  "}ul.lst-kix_hefi1acv3kbb-4{list-style-type:none}ul.lst-kix_hefi1acv3kbb-5{list-style-type:none}ul.lst-kix_hefi1acv3kbb-2{list-style-type:none}ul.lst-kix_hefi1acv3kbb-3{list-style-type:none}.lst-kix_d0pnlbhmephp-8>li:before{content:"\0025a0  "}ul.lst-kix_90sjlybkt4o7-8{list-style-type:none}ul.lst-kix_90sjlybkt4o7-7{list-style-type:none}ul.lst-kix_90sjlybkt4o7-6{list-style-type:none}ul.lst-kix_90sjlybkt4o7-5{list-style-type:none}.lst-kix_d0pnlbhmephp-2>li:before{content:"\0025a0  "}.lst-kix_d0pnlbhmephp-4>li:before{content:"\0025cb  "}ul.lst-kix_90sjlybkt4o7-4{list-style-type:none}ul.lst-kix_90sjlybkt4o7-3{list-style-type:none}ul.lst-kix_90sjlybkt4o7-2{list-style-type:none}.lst-kix_d0pnlbhmephp-0>li:before{content:"\0025cf  "}ul.lst-kix_90sjlybkt4o7-1{list-style-type:none}.lst-kix_vknp6nujnm9e-1>li:before{content:"\0025cb  "}ul.lst-kix_90sjlybkt4o7-0{list-style-type:none}.lst-kix_nh8twm5hcw9o-6>li:before{content:"\0025cf  "}.lst-kix_cbfpvidehrpt-5>li:before{content:"\0025a0  "}.lst-kix_hefi1acv3kbb-0>li:before{content:"\0025cf  "}.lst-kix_yoy1ormm4wf-2>li:before{content:"\0025a0  "}.lst-kix_yoy1ormm4wf-0>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-6>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-4>li:before{content:"\0025cb  "}.lst-kix_hefi1acv3kbb-8>li:before{content:"\0025a0  "}ul.lst-kix_hefi1acv3kbb-8{list-style-type:none}.lst-kix_hefi1acv3kbb-2>li:before{content:"\0025a0  "}ul.lst-kix_hefi1acv3kbb-6{list-style-type:none}ul.lst-kix_hefi1acv3kbb-7{list-style-type:none}.lst-kix_yoy1ormm4wf-4>li:before{content:"\0025cb  "}.lst-kix_yoy1ormm4wf-6>li:before{content:"\0025cf  "}.lst-kix_yoy1ormm4wf-8>li:before{content:"\0025a0  "}.lst-kix_weixxh5ko91p-2>li:before{content:"\0025a0  "}ul.lst-kix_ahj95ozgae0m-0{list-style-type:none}ul.lst-kix_ahj95ozgae0m-2{list-style-type:none}ul.lst-kix_ahj95ozgae0m-1{list-style-type:none}.lst-kix_90sjlybkt4o7-8>li:before{content:"\0025a0  "}.lst-kix_90sjlybkt4o7-7>li:before{content:"\0025cb  "}ul.lst-kix_yoy1ormm4wf-8{list-style-type:none}.lst-kix_u5mt6v244o5o-5>li:before{content:"\0025a0  "}.lst-kix_u5mt6v244o5o-7>li:before{content:"\0025cb  "}ul.lst-kix_yoy1ormm4wf-7{list-style-type:none}.lst-kix_u5mt6v244o5o-6>li:before{content:"\0025cf  "}.lst-kix_90sjlybkt4o7-3>li:before{content:"\0025cf  "}.lst-kix_u5mt6v244o5o-1>li:before{content:"\0025cb  "}.lst-kix_90sjlybkt4o7-2>li:before{content:"\0025a0  "}.lst-kix_u5mt6v244o5o-0>li:before{content:"\0025cf  "}.lst-kix_u5mt6v244o5o-8>li:before{content:"\0025a0  "}.lst-kix_90sjlybkt4o7-0>li:before{content:"\0025cf  "}.lst-kix_90sjlybkt4o7-1>li:before{content:"\0025cb  "}.lst-kix_u5mt6v244o5o-2>li:before{content:"\0025a0  "}.lst-kix_u5mt6v244o5o-3>li:before{content:"\0025cf  "}.lst-kix_u5mt6v244o5o-4>li:before{content:"\0025cb  "}.lst-kix_ahj95ozgae0m-4>li:before{content:"\0025cb  "}ul.lst-kix_cbfpvidehrpt-3{list-style-type:none}ul.lst-kix_cbfpvidehrpt-2{list-style-type:none}ul.lst-kix_cbfpvidehrpt-5{list-style-type:none}ul.lst-kix_cbfpvidehrpt-4{list-style-type:none}.lst-kix_ahj95ozgae0m-1>li:before{content:"\0025cb  "}.lst-kix_ahj95ozgae0m-5>li:before{content:"\0025a0  "}ul.lst-kix_cbfpvidehrpt-1{list-style-type:none}ul.lst-kix_cbfpvidehrpt-0{list-style-type:none}.lst-kix_ahj95ozgae0m-0>li:before{content:"\0025cf  "}.lst-kix_ahj95ozgae0m-8>li:before{content:"\0025a0  "}.lst-kix_ahj95ozgae0m-6>li:before{content:"\0025cf  "}.lst-kix_ahj95ozgae0m-7>li:before{content:"\0025cb  "}ul.lst-kix_wulsapsxo6ef-2{list-style-type:none}ul.lst-kix_wulsapsxo6ef-3{list-style-type:none}ul.lst-kix_wulsapsxo6ef-0{list-style-type:none}ul.lst-kix_wulsapsxo6ef-1{list-style-type:none}ul.lst-kix_wulsapsxo6ef-6{list-style-type:none}ul.lst-kix_wulsapsxo6ef-7{list-style-type:none}ul.lst-kix_wulsapsxo6ef-4{list-style-type:none}ul.lst-kix_wulsapsxo6ef-5{list-style-type:none}.lst-kix_90sjlybkt4o7-4>li:before{content:"\0025cb  "}ul.lst-kix_ahj95ozgae0m-8{list-style-type:none}ul.lst-kix_u5mt6v244o5o-7{list-style-type:none}ul.lst-kix_yoy1ormm4wf-2{list-style-type:none}ul.lst-kix_ahj95ozgae0m-7{list-style-type:none}ul.lst-kix_u5mt6v244o5o-8{list-style-type:none}ul.lst-kix_yoy1ormm4wf-1{list-style-type:none}.lst-kix_90sjlybkt4o7-5>li:before{content:"\0025a0  "}ul.lst-kix_yoy1ormm4wf-0{list-style-type:none}ul.lst-kix_wulsapsxo6ef-8{list-style-type:none}.lst-kix_90sjlybkt4o7-6>li:before{content:"\0025cf  "}.lst-kix_ahj95ozgae0m-2>li:before{content:"\0025a0  "}ul.lst-kix_ahj95ozgae0m-4{list-style-type:none}ul.lst-kix_u5mt6v244o5o-3{list-style-type:none}ul.lst-kix_yoy1ormm4wf-6{list-style-type:none}ul.lst-kix_cbfpvidehrpt-7{list-style-type:none}ul.lst-kix_ahj95ozgae0m-3{list-style-type:none}ul.lst-kix_u5mt6v244o5o-4{list-style-type:none}ul.lst-kix_yoy1ormm4wf-5{list-style-type:none}ul.lst-kix_cbfpvidehrpt-6{list-style-type:none}.lst-kix_ahj95ozgae0m-3>li:before{content:"\0025cf  "}ul.lst-kix_ahj95ozgae0m-6{list-style-type:none}ul.lst-kix_u5mt6v244o5o-5{list-style-type:none}ul.lst-kix_yoy1ormm4wf-4{list-style-type:none}ul.lst-kix_ahj95ozgae0m-5{list-style-type:none}ul.lst-kix_u5mt6v244o5o-6{list-style-type:none}ul.lst-kix_yoy1ormm4wf-3{list-style-type:none}ul.lst-kix_cbfpvidehrpt-8{list-style-type:none}.lst-kix_cbfpvidehrpt-6>li:before{content:"\0025cf  "}ul.lst-kix_u5mt6v244o5o-0{list-style-type:none}.lst-kix_nh8twm5hcw9o-5>li:before{content:"\0025a0  "}ul.lst-kix_u5mt6v244o5o-1{list-style-type:none}ul.lst-kix_d0pnlbhmephp-0{list-style-type:none}ul.lst-kix_u5mt6v244o5o-2{list-style-type:none}.lst-kix_cbfpvidehrpt-4>li:before{content:"\0025cb  "}.lst-kix_cbfpvidehrpt-8>li:before{content:"\0025a0  "}.lst-kix_nh8twm5hcw9o-3>li:before{content:"\0025cf  "}.lst-kix_nh8twm5hcw9o-7>li:before{content:"\0025cb  "}.lst-kix_cbfpvidehrpt-2>li:before{content:"\0025a0  "}ul.lst-kix_d0pnlbhmephp-6{list-style-type:none}.lst-kix_d0pnlbhmephp-7>li:before{content:"\0025cb  "}.lst-kix_nh8twm5hcw9o-1>li:before{content:"\0025cb  "}ul.lst-kix_d0pnlbhmephp-5{list-style-type:none}ul.lst-kix_d0pnlbhmephp-8{list-style-type:none}ul.lst-kix_d0pnlbhmephp-7{list-style-type:none}.lst-kix_weixxh5ko91p-5>li:before{content:"\0025a0  "}.lst-kix_wulsapsxo6ef-0>li:before{content:"\0025cf  "}ul.lst-kix_d0pnlbhmephp-2{list-style-type:none}.lst-kix_d0pnlbhmephp-5>li:before{content:"\0025a0  "}ul.lst-kix_d0pnlbhmephp-1{list-style-type:none}ul.lst-kix_d0pnlbhmephp-4{list-style-type:none}ul.lst-kix_d0pnlbhmephp-3{list-style-type:none}.lst-kix_weixxh5ko91p-7>li:before{content:"\0025cb  "}.lst-kix_cbfpvidehrpt-0>li:before{content:"\0025cf  "}ul.lst-kix_weixxh5ko91p-8{list-style-type:none}ul.lst-kix_weixxh5ko91p-7{list-style-type:none}ul.lst-kix_weixxh5ko91p-6{list-style-type:none}ul.lst-kix_weixxh5ko91p-5{list-style-type:none}.lst-kix_d0pnlbhmephp-3>li:before{content:"\0025cf  "}ul.lst-kix_weixxh5ko91p-0{list-style-type:none}ul.lst-kix_weixxh5ko91p-4{list-style-type:none}.lst-kix_d0pnlbhmephp-1>li:before{content:"\0025cb  "}ul.lst-kix_weixxh5ko91p-3{list-style-type:none}ul.lst-kix_weixxh5ko91p-2{list-style-type:none}.lst-kix_vknp6nujnm9e-2>li:before{content:"\0025a0  "}ul.lst-kix_weixxh5ko91p-1{list-style-type:none}.lst-kix_vknp6nujnm9e-0>li:before{content:"\0025cf  "}.lst-kix_yoy1ormm4wf-3>li:before{content:"\0025cf  "}.lst-kix_yoy1ormm4wf-1>li:before{content:"\0025cb  "}.lst-kix_hefi1acv3kbb-7>li:before{content:"\0025cb  "}.lst-kix_hefi1acv3kbb-3>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-1>li:before{content:"\0025cb  "}.lst-kix_yoy1ormm4wf-5>li:before{content:"\0025a0  "}.lst-kix_yoy1ormm4wf-7>li:before{content:"\0025cb  "}.lst-kix_weixxh5ko91p-3>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-5>li:before{content:"\0025a0  "}.lst-kix_weixxh5ko91p-1>li:before{content:"\0025cb  "}ol{margin:0;padding:0}table td,table th{padding:0}.c0{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:normal}.c1{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left;height:11pt}.c15{-webkit-text-decoration-skip:none;color:#000000;text-decoration:underline;vertical-align:baseline;text-decoration-skip-ink:none;font-family:"Arial";font-style:normal}.c4{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-family:"Arial";font-style:normal}.c28{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:right}.c3{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c22{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:center}.c9{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Arial"}.c13{color:#000000;text-decoration:none;vertical-align:baseline;font-style:normal}.c21{color:#000000;text-decoration:none;vertical-align:baseline}.c26{background-color:#ffffff;max-width:511.2pt;padding:50.4pt 50.4pt 50.4pt 50.4pt}.c16{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;text-decoration:underline}.c8{margin-left:36pt;padding-left:0pt}.c7{margin-left:72pt;padding-left:0pt}.c11{color:inherit;text-decoration:inherit}.c10{font-family:"Courier New";font-weight:400}.c6{padding:0;margin:0}.c17{margin-left:108pt;padding-left:0pt}.c27{font-size:11pt;font-style:normal}.c18{color:#1155cc}.c24{height:11pt}.c20{text-indent:36pt}.c14{font-weight:700}.c25{margin-left:36pt}.c23{background-color:#cccccc}.c12{background-color:#ffff00}.c2{font-size:12pt}.c5{font-style:italic}.c19{font-weight:400}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body><div class="c26"><p class="c22"><span class="c4 c2">SlimShadey: Java based creation, visualization, and editing of multiple sequence alignments</span></p><p class="c22"><span class="c14 c2">User </span><span class="c4 c2">Manual</span></p><p class="c22"><span class="c0">Avi Shah and Sudhir Nayak</span></p><p class="c22"><span class="c0">The College of New Jersey</span></p><p class="c22 c24"><span class="c15 c14 c2"></span></p><p class="c3"><span class="c4 c2">Contents</span></p><p class="c3"><span class="c16 c18 c2"><a class="c11" href="#id.paxlo16wvvi1">Introduction (2)</a></span></p><p class="c3 c25"><span class="c16 c18 c2"><a class="c11" href="#id.1l89yc7mrcp0">What is SlimShadey?</a></span></p><p class="c3 c25"><span class="c16 c18 c2"><a class="c11" href="#id.hbtfivvg4dv8">Requirements</a></span><hr style="page-break-before:always;display:none;"></p><p class="c1"><span class="c15 c14 c2"></span></p><a id="id.paxlo16wvvi1"></a><p class="c3"><span class="c15 c14 c2">Introduction</span></p><a id="id.1l89yc7mrcp0"></a><p class="c3"><span class="c4 c2">What is Slimshadey?</span></p><p class="c3"><span class="c2">SlimShadey is a light-weight Java-based tool that allows for direct residue level control in the editing, trimming, shading, and export of multiple sequence alignments. The extensive functionality implemented in SlimShadey takes advantage of JavaFX and includes complete editing (inserting, deleting, degapping, subsequence export), real-time shading (identity, similarity, dissimilarity, counts, matrix), annotation (domains, motifs, logos), and exporting of publication quality images in PNG and RTF format. &nbsp;This manual and </span><span class="c2 c12">associated videos</span><span class="c0">&nbsp;use the GLD-1 or CD44 protein sequences (reviewed in Lee and Schedl 2010 and Senbanjo and Chellaiah, 2017, respectively) to illustrate the major tools contained in the package. &nbsp;Additional sample sequences are provided to illustrate the range sequence alignments that can be manipulated. &nbsp;All of the calculation, shading, and editing of sample sequences is performed in real-time.</span></p><p class="c1"><span class="c0"></span></p><a id="id.hbtfivvg4dv8"></a><p class="c3"><span class="c4 c2">Requirements</span></p><p class="c3"><span class="c2">SlimShadey will run without modification on Windows 7 (or higher), OS X / macOS, and Linux operating systems when Java 8 JRE or higher is installed (download from </span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.oracle.com/java/technologies/javase-jre8-downloads.html&amp;sa=D&amp;ust=1603771834437000&amp;usg=AOvVaw0-QsgdyBq2Y2eu790IhZTE">Oracle</a></span><span class="c0">). &nbsp;SlimShadey has minimal RAM requirements (2GB), does not require a dedicated graphics card, and will run on legacy systems provided they support Java 8 JRE. &nbsp;However, for larger datasets and a smoother experience, the availability of 16GB RAM (or higher), dedicated graphics card (for example, NVIDIA GTX 1060), and quad core processor or higher (for example, Intel Core i7-8750) will significantly improve performance. Internet access is not required for SlimShadey, except for ScanProsite- and hmmscan-based shading.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c15 c2 c19">Checks to run before starting</span></p><p class="c3"><span class="c0">On your PC or server, from the terminal check your Java using &ldquo;java -version&rdquo;. &nbsp;This should return something similar to the the following:</span></p><p class="c3 c20"><span class="c13 c10 c2">java version &quot;1.8.0_251&quot;</span></p><p class="c3 c20"><span class="c13 c10 c2">Java(TM) SE Runtime Environment (build 1.8.0_251-b08)</span></p><p class="c3 c20"><span class="c13 c10 c2">Java HotSpot(TM) 64-Bit Server VM (build 25.251-b08, mixed mode)</span></p><p class="c3"><span class="c0">If you meet or exceed the requirements above then SlimShadey will operate on your system.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">In addition, SlimShadey will run most server platforms that support Java. &nbsp;From the terminal check your Java using &ldquo;java -version&rdquo; and it should be the same as the output above. &nbsp;If your server does not meet the minimum requirements for SlimShadey, please contact your system administrator.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Input file formats</span></p><p class="c3"><span class="c2">The preferred input file format for SlimShadey is sequential FASTA (aligned or unaligned). Alignments in other formats can easily be converted using </span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/&amp;sa=D&amp;ust=1603771834438000&amp;usg=AOvVaw3rrSDMDPR6MZSJbPzP6HcV">SeqRet</a></span><span class="c2">&nbsp;(Madeira et al, 2019). &nbsp;For unaligned sequences, </span><span class="c0">SlimShadey has been packaged with OpalFX (Wheeler and Kececioglu, 2007) to allow for the rapid alignment of protein and nucleic acid sequences. SlimShadey supports the direct cut/paste of FASTA files from NCBI, Ensembl, and other databases.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">Example input format (partial GLD-1 FASTA OPAL aligned):</span></p><p class="c1"><span class="c10 c2 c5 c21"></span></p><p class="c3"><span class="c13 c10 c2">&gt;GLD-1_Caenorhabditis_latens</span></p><p class="c3"><span class="c13 c10 c2">FGVTSQLETQNAESPSRSSILTPTSLEDETSPRKRFPLIETNISASDRWPPAPRRDGWSS</span></p><p class="c1"><span class="c13 c10 c2"></span></p><p class="c3"><span class="c13 c10 c2">&gt;GLD-1_Caenorhabditis_briggsae</span></p><p class="c3"><span class="c13 c10 c2">FGVSAQHENPSVDSPSRSSILTPTSLDDETSPRKSFQILESSVS-ADRWP-APRRDGWSS</span></p><p class="c1"><span class="c13 c10 c2"></span></p><p class="c3"><span class="c13 c10 c2">&gt;GLD-1_Caenorhabditis_nigoni</span></p><p class="c3"><span class="c13 c10 c2">FGVSAQHENPSVDSPSRSSILTPTSLDDETSPRKGFPILESSVS-TDRWP-APRRDGWSS</span></p><p class="c1"><span class="c13 c10 c2"></span></p><p class="c3"><span class="c0">SlimShadey also accepts the Shadey project file format (extension .mm), which stores all sequence, annotation, consensus, and shading/styling information. &nbsp;This allows users to share their alignment markup/workflow and preferred session settings, to work on an alignment collaboratively or over multiple SlimShadey sessions.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2 c5">Input and output formats </span></p><p class="c3"><span class="c0">Other accepted input and output formats include Clustal, MSF, and Phylup 3.2. &nbsp;SlimShadey will extract the text, convert it to a FASTA file, and will discard any additional information. SlimShadey can also output to Clustal, MSF, and Phylup 3.2; however, only the sequence and sequence names will be saved (annotation and shading will be lost). </span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2">Sequence names are appended to 10 characters when outputting to MSF and Phylup 3.2. The sequence names remain the same when outputting to FASTA and Clustal sequence names are appended to 20 characters.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">SlimShadey Format (.mm)</span></p><p class="c3"><span class="c0">The .mm format has been designed to hold all of the information from a SlimShadey project. &nbsp; &nbsp;The text file holds all sequence information plus annotations and complete shading (including position, color, shade) information.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">File structure:</span></p><ul class="c6 lst-kix_u5mt6v244o5o-0 start"><li class="c3 c8"><span class="c2">The first 5 characters of a </span><span class="c10 c2">.mm</span><span class="c2">&nbsp;file are &ldquo;</span><span class="c10 c2">.slim</span><span class="c0">&rdquo;.</span></li><li class="c3 c8"><span class="c2">Data fields begin with &ldquo;</span><span class="c10 c2">$</span><span class="c2">&rdquo; followed by &ldquo;</span><span class="c10 c2">field_name</span><span class="c2">&rdquo; (e.g. </span><span class="c10 c2">fontsize</span><span class="c2">, </span><span class="c10 c2">collapse</span><span class="c2">, </span><span class="c10 c2">luminance</span><span class="c2">, etc.), and, within curly brackets (&ldquo;</span><span class="c10 c2">{</span><span class="c2">&ldquo; and &ldquo;</span><span class="c10 c2">}</span><span class="c2">&rdquo;) the data for that field. For example, to denote that the given alignment&rsquo;s UI font size is 24, a line in the file would be: </span><span class="c10 c2">$fontsize{24}</span></li></ul><ul class="c6 lst-kix_wulsapsxo6ef-0 start"><li class="c3 c8"><span class="c0">Sequences and annotations are stored with one character per line, and separated by sequence names. For example, valid sequence data for the following FASTA alignment:</span></li></ul><p class="c3"><span class="c13 c10 c2">&gt;seq1</span></p><p class="c3"><span class="c13 c10 c2">AY</span></p><p class="c3"><span class="c13 c10 c2">&gt;seq2</span></p><p class="c3"><span class="c10 c2">AW</span></p><ul class="c6 lst-kix_vknp6nujnm9e-0 start"><li class="c3 c8"><span class="c0">With one empty (i.e. all space characters) annotation row, and shaded with white background and black text, the alignment would be saved as:</span></li></ul><p class="c3"><span class="c13 c10 c2">$annotation_data{</span></p><p class="c3"><span class="c13 c10 c2">%EMPTY_ANNO_1</span></p><p class="c3"><span class="c13 c10 c2">&nbsp;.#FFFFFF.#000000</span></p><p class="c3"><span class="c13 c10 c2">&nbsp;.#FFFFFF.#000000</span></p><p class="c3"><span class="c13 c10 c2">}</span></p><p class="c3"><span class="c13 c10 c2">$sequence_data{</span></p><p class="c3"><span class="c13 c10 c2">%seq1</span></p><p class="c3"><span class="c13 c10 c2">A.#FFFFFF.#000000</span></p><p class="c3"><span class="c13 c10 c2">Y.#FFFFFF.#000000</span></p><p class="c3"><span class="c13 c10 c2">%seq2</span></p><p class="c3"><span class="c13 c10 c2">A.#FFFFFF.#000000</span></p><p class="c3"><span class="c13 c10 c2">W.#FFFFFF.#000000</span></p><p class="c3"><span class="c13 c10 c2">}</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Sample data provided</span></p><p class="c3"><span class="c0">The figures used in this manual are screen captures from the following video:</span></p><p class="c3"><span class="c0 c12">Link to video</span></p><p class="c1"><span class="c0 c12"></span></p><p class="c3"><span class="c4 c2">Parameter file</span></p><p class="c3"><span class="c2">SlimShadey includes a parameter file that can be easily modified with a text editor to change default rules for each alphabet without the need for recompiling. &nbsp;For example, new amino acid groups can be defined, the matrices can be modified or new ones added, consensus rules changed, default coloring modified, and non-standard characters can be added. &nbsp;While default coloring RGB values (in hex format) and consensus rules can be changed without any issues, if you add a new character, it must be represented in each substitution matrix. &nbsp;</span><span class="c0 c12">For a more detailed description see &hellip;&hellip;&hellip;.. And the following video.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Unzip the .jar file, and go to the sub-directory:</span></p><p class="c3"><span class="c13 c10 c2">[edu/tcnj/seqverter/sequence/]</span></p><p class="c3"><span class="c2">In the </span><span class="c10 c2">[.../sequence/] </span><span class="c2">directory, there are </span><span class="c10 c2">*.salpha</span><span class="c2">&nbsp;files, which contain the rules. For example, to change the rules for protein sequences, the user can edit the </span><span class="c10 c2">Protein.salpha</span><span class="c0">&nbsp;file.</span></p><p class="c1"><span class="c15 c14 c2"></span></p><p class="c3"><span class="c15 c14 c2">File menu options</span></p><p class="c3"><span class="c4 c2">Rename tab</span></p><p class="c3"><span class="c0">Allows the user to rename the selected tab. Custom tab names allow for streamlining of workflow, by specifying the contents of each open alignment, and giving a default filename when the project is exported to FASTA, RTF, PNG, or the Shadey project format. &nbsp;Each tab is treated as a different project and must be saved separately. &nbsp;The number of tabs that can be open during a session is limited by total memory allocated to the JVM (see troubleshooting). &nbsp;</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Realign sequences (align sequences)</span></p><p class="c3"><span class="c0">SlimShadey is provided with OpalFX, an interface to the Opal multiple sequence alignment algorithm (Wheeler, 2007), as a fast, accurate, and convenient sequence alignment tool. &nbsp;OpalFX can be used to align unaligned sequences and to realign a multiple sequence alignment already loaded into a tab. OpalFX also supports translation and alignment of nucleotide sequences with all 30+ known genetic codes, as well as codon-delimited alignment. &nbsp;Once alignments have been completed, the user can directly import the results to SlimShadey or save the alignment in FASTA format from the file menu.</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">Print preview</span></p><p class="c3"><span class="c0">Print preview allows for WYSIWYG formatting and rendering of publication quality images. The multiple sequence alignment is organized into rows, and the user is given control over numbering, truncation of names, font size, number of columns per row, and whether to include annotations or a consensus sequence. The output can be saved as a PNG file for downstream analysis for further annotation in image processing software. &nbsp;</span></p><p class="c1"><span class="c9 c2 c5"></span></p><ul class="c6 lst-kix_90sjlybkt4o7-0 start"><li class="c3 c8"><span class="c2 c5">NOTE: </span><span class="c0">PNG resolution is based on font size. Increasing the font size will increase the output resolution. As such, if additional resolution is necessary then font size can be adjusted to meet your requirements.</span></li></ul><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Output to RTF (rich text format)</span></p><p class="c3"><span class="c0">Exporting the contents of a tab in RTF format allows the output of SlimShadey to be opened and edited manually in word processing software (e.g. MS Word, OpenOffice, Google Docs). Exporting to RTF allows users to directly paste a styled/formatted alignment into other documents. This format also allows the user to change (add, insert, and delete) characters after the contents of a tab have been output without using the SlimShadey interface. &nbsp;As output options, the user is provided control over font size, name truncation (i.e. number of characters in name), and whether to include annotations or numbering.</span></p><p class="c1"><span class="c0"></span></p><ul class="c6 lst-kix_d0pnlbhmephp-0 start"><li class="c3 c8"><span class="c2 c5">NOTE: </span><span class="c0">The RTF is saved with all color and formatting specified exactly as in the SlimShadey tab. Some word processors may not display the RTF properly. For best results we recommend Microsoft Word.</span></li></ul><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Save as FASTA</span></p><p class="c3"><span class="c2">SlimShadey allows any tab to be exported to FASTA format for use in other applications. &nbsp;Other formats can be generated using </span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/&amp;sa=D&amp;ust=1603771834445000&amp;usg=AOvVaw2cbFijlQsYVvngbig7LXY_">SeqRet</a></span><span class="c2">&nbsp;(Madeira et al, 2019).</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c14 c2">Save as project</span></p><p class="c3"><span class="c2">SlimShadey outputs projects to the .mm format, allowing complete sessions to be shared. The &ldquo;</span><span class="c2 c5">.mm&rdquo; </span><span class="c0">format saves the alignment along with all shading, annotations and consensus sequence, session settings, including UI font size, the first index of the alignment, luminance threshold for text coloring, the alphabet the alignment (RNA, DNA, or protein), and whether to collapse identical sequences.</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c16 c14 c2">Edit menu options</span></p><p class="c3"><span class="c4 c2">Session settings</span></p><ul class="c6 lst-kix_hefi1acv3kbb-0 start"><li class="c3 c8"><span class="c2 c5">First index:</span><span class="c0">&nbsp;Sets the index of the first column of the alignment to the specified value. By default, the first column is &lsquo;1&rsquo;. &nbsp;For example, if you have a portion of a larger sequence, you may want to change this value to better reflect the position of the sequence.</span></li><li class="c3 c8"><span class="c2 c5">Font size:</span><span class="c0">&nbsp;Changes the UI font size to the specified value. Font size is 18px by default, to display as much of the alignment as clearly as possible on most machines. &nbsp;SlimShadey renders all characters in real-time so changes in font size will change the number of characters displayed on the screen and may alter the performance of the UI.</span></li><li class="c3 c8"><span class="c2 c5">Luminance threshold: </span><span class="c0">&nbsp;The luminance of a color can be calculated from its RGB color space. Depending on whether the shading color&rsquo;s luminance is above or below the threshold value, the text color is set to either black or white. The user can change the text color to any RGB value manually, but built-in shading/coloring algorithms will only color text black or white when shading is performed, depending on the background shading assigned. After shading is performed, the user has residue-level control over the color of the letter and the background. &nbsp;The luminance threshold is by default 0.179, which generally assigns white or black text color appropriately based on the background shading color. Changing this value is not recommended.</span></li></ul><p class="c1"><span class="c14 c2 c15"></span></p><p class="c3"><span class="c4 c2">Gap only columns</span></p><p class="c3"><span class="c0">SlimShadey can identify gap only columns so they can be removed by the user or avoided prior to exporting subsequences.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Calculate consensus</span></p><p class="c3"><span class="c0">SlimShadey can calculate three types of consensus sequence (Simple, Dots and Identify, and ClustalX). &nbsp;Simple consensus outputs a lowercase character for majority in a column and an uppercase when characters are identical in a column. Dots and identify uses the ClustalW rule set to calculate the consensus (Thompson, 1994). ClustalX consensus uses the rule set from Jeanmougin et al., 1998.</span></p><p class="c1"><span class="c0 c12"></span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0 c12">ClustalX rules (Jeanmougin, 1998)</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">Add annotation row</span></p><p class="c3"><span class="c2">This function allows the user to add an annotation row. For detailed instructions on editing, moving, deleting, or adding annotations, please refer to the section on </span><span class="c2 c14">Directly Editing the Alignment</span><span class="c2">. Annotation rows are useful for highlighting features of the alignment, such as regions with conserved structure, outstanding residues, insertions, or deletions of interest. Annotations can be added by individual column or by region (block annotation over multiple columns). &nbsp;For an annotation over a single column, right-click above the region of interest and select &ldquo;Annotate selection&rdquo; from the pull-down menu. &nbsp;If you want to annotate a region, right-click-and-drag over the region desired and select &ldquo;Annotate selection&rdquo;. &nbsp;Commonly used characters are available for copy/paste use in the annotation window. &nbsp;For regional annotation the single character will be copied over all columns. &nbsp;For example, to define an alpha helical region, the user can cut-paste the &lsquo;</span><span class="c10 c2">&alpha;</span><span class="c2">&rsquo;, which will be copied over all columns selected. &nbsp;If multiple characters are used then the characters will repeat over the region, e.g if 4 columns are selected, and &lsquo;</span><span class="c10 c2">&alpha;</span><span class="c2">&rsquo; is entered, the annotation &lsquo;</span><span class="c10 c2">&alpha;&alpha;&alpha;&alpha;</span><span class="c2">&rsquo; will be applied. &nbsp;</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">Collapse identical sequences</span></p><p class="c3"><span class="c0">This function collapses identical sequences into groups, which can be unpacked when analysis has been completed without any loss of data. This function works by performing an all-against-all comparison for identify and only displays unique sequences. When a group of identical sequences is collapsed into a group, only the shading and text color of the first sequence is saved; when uncollapsing the sequences, the shading that was applied to the collapsed sequence is applied to all constituent sequences.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">SlimShadey performs real-time calculations on the alignment as rendered. &nbsp;For example, if there are multiple identical sequences that result in the over representation of characters, when the collapse function is called and they are reduced to a single sequence, the same characters may now be underrepresented. &nbsp;More specifically, if you have 5 sequences, 3 of which are identical, then the 3 identical sequences will be a majority when uncollapsed, and will not be a majority after collapsing.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">When working with large datasets that may have many identical sequences, using this function can improve the performance of SlimShadey. For example, rapidly identifying low frequency polymorphisms in a population benefits from not rendering uninformative sequences. &nbsp;</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c15 c14 c2">Shade menu options</span></p><p class="c3"><span class="c4 c2">Shade by unique residues</span></p><p class="c3"><span class="c2">To reveal outstanding residues, this function highlights unique residues in a column. &nbsp;For flexibility, the remote allows adjusting the maximum number of residues in the column. &nbsp;This allows restricting shading to those residues within a column whose count is below a threshold count. &nbsp;This is an easy and sensitive method for determining interesting polymorphisms in sequence alignments. The ability to store logs of differing residues by position, and visualize the locations and apparent densities of these differences, may be helpful in directing further analysis and generating testable hypotheses.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Shade by residue frequency</span></p><p class="c3"><span class="c0">Within a column, if the proportion of a single residue exceeds this threshold frequency, the residue is shaded black to denote identity, and similar residues gray to denote similarity; if the total proportion of different, but physically or chemically similar residues exceeds the threshold, those similar residues are shaded gray to denote similarity (e.g. if in a column neither leucine nor isoleucine alone exceed the threshold frequency, but together do, both are shaded &lsquo;gray&rsquo;). A threshold between 0.5 and 1 can be set by the user. The lower limit of 0.5 is set to avoid determining more than one consensus residue within a column. &nbsp;</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2">This color scheme can be inverted (Inverse shading) to produce &lsquo;inverse shading,&rsquo; in which residues to be shaded black are left white, residues to be left white are shaded black, and residues to be shaded grey are shaded grey; color inversion is particularly useful for demonstrating dissimilarity or lack of conservation.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Shade by sequence</span></p><p class="c3"><span class="c0">Shade by sequence allows for shading matches, similarities, and differences relative to a master sequence selected by the user. &nbsp;The master sequence can be any of the sequences in the alignment. &nbsp;By default, the master sequence is not shaded; however, this can be changed to any color from the remote. &nbsp;This function is useful for revealing identity to a specific lineage.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Prosite-based shading</span></p><p class="c3"><span class="c2">Prosite-based shading retrieves matches from prosite and renders them on the alignment (Hulo, 2004; Sigrist, 2013). &nbsp;The default implementation uses ExPASy&rsquo;s standard settings and filters low complexity patterns and includes ProSite profiles in the output. &nbsp;The hits can be applied to the alignment, where all related domains will be rendered in the same color. &nbsp;Information about the pattern, including ScanProsite hit documentation, can be retrieved by clicking on the colored region of the diagram. &nbsp;Sequences are degapped prior to scanning so that hits will be rendered even if there are gaps or the hits are present at different positions.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Default RASMOL shading</span></p><p class="c3"><span class="c0">Default coloring of alignments uses the RASMOL color scheme (Sayle et al, 1995), which uses contrasting colors to distinguish residues with different physico-chemical properties.</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">Default ClustalX shading</span></p><p class="c3"><span class="c2">This function calculates a consensus and applies shading using the rules from ClustalX for protein sequences (Larkin, 2007). &nbsp;Each amino acid is shaded according to its identity, and the consensus character for its column. The rules for this are described in the &lsquo;</span><span class="c10 c2">colorprot.par</span><span class="c0">&rsquo; parameter file distributed with ClustalX.</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">Shade with substitution matrix</span></p><p class="c3"><span class="c0">Sequences can be shaded with a relative match to one of the BLOSUM matrix series (30-100), Gonnet250, PAM250, or identity. &nbsp;For each substitution matrix, a shading matrix is calculated based on two RGB values, one for a perfect match and one for the worst possible match. For each residue, this corresponds to the maximum (i.e. identity) and minimum score possible, respectively. For example, when using BLOSUM62, and comparing a residue to alanine (A), the maximum score is when aligned to another A, at +4. &nbsp;This represents a perfect match and is assigned the match color. &nbsp;Similarly, using BLOSUM62, alanine aligned with tryptophan (W) results in the minimum score of -3, and produces shading closest to the mismatch color. &nbsp;As the residue changes from the worst possible match, and the score of the aligned residue increases, the RGB values scales linearly to the match color. Comparisons to a gap character (-) always result in no shading.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Shade &nbsp;to a sequence</span></p><p class="c3"><span class="c2">SlimShadey allows shading relative to a specific sequence in the alignment or to a frequency-based consensus. &nbsp;</span><span class="c0">Shading to a sequence allows the user to define a &lsquo;master sequence&rsquo; as the template for shading. This master sequence is optionally shaded to the match color, and every other sequence is compared column-wise to the master-sequence to determine shading, as described above. For example, identity to a single sequence in the alignment can be shaded by selecting the sequence from the &ldquo;Shade to shade to&rdquo; pull-down menu and the identity matrix from the &ldquo;Substitution matrix&rdquo; pull-down menu.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Shade to a simple majority</span></p><p class="c3"><span class="c0">Shading to a simple majority defines a frequency-based consensus as the master sequence and shading is performed as described above. The threshold frequency for calculating the consensus can be adjusted using the slider, to between 0.5 and 1, to change shading in real-time.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Shade with HMM</span></p><p class="c3"><span class="c2">The hmmer sequence analysis package implements an algorithm for finding motifs and domains within sequences using profile HMMs (Potter et al., 2018). SlimShadey uses an online implementation of hmmscan, hosted at the EBI (</span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.ebi.ac.uk/Tools/hmmer/search/hmmscan&amp;sa=D&amp;ust=1603771834452000&amp;usg=AOvVaw3VtHXM7JY2zJo4ums-w3Un">https://www.ebi.ac.uk/Tools/hmmer/search/hmmscan</a></span><span class="c0">) to identify motifs in each sequence in the alignment. Motifs are all from the Pfam database (Potter et al., 2018), and documentation can be retrieved for these within SlimShadey. By default, all hits to the same accession are assigned the same RGB value for shading. &nbsp;Each motif&rsquo;s default color can be changed independently, and individual motifs can be shown or hidden on the alignment. &nbsp;All positional information is carried from hmmscan to SlimShadey and retained on the alignment, even when gaps are present within a motif.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c4 c2">Shade with SwissProt regular expressions</span></p><p class="c3"><span class="c2">SlimShadey allows shading of sequences using matches to regular expressions using SwissProt&rsquo;s ScanProsite syntax. &nbsp;The SwissProt regular expression syntax can be downloaded </span><span class="c16 c18 c2"><a class="c11" href="ftp://ftp.expasy.org/databases/prosite/profile.txt">here</a></span><span class="c2">&nbsp;or viewed </span><span class="c16 c2 c18"><a class="c11" href="https://www.google.com/url?q=https://prosite.expasy.org/scanprosite/scanprosite_doc.html%23mo_motifs&amp;sa=D&amp;ust=1603771834453000&amp;usg=AOvVaw31ht07cSYD2tKbK0h8KC9_">here</a></span><span class="c2">. Each sequence in the alignment is scanned for a match to the pattern, regardless of position, and matches are previewed in a graphical window. &nbsp;Any matches can be applied to the alignment and viewed directly on the sequence alignment. &nbsp;SlimShadey&rsquo;s scanner disregards gap characters, e.g. the pattern &lsquo;</span><span class="c10 c23 c2">MP</span><span class="c2">&rsquo; will match a subsequence given by &lsquo;</span><span class="c2 c10">...AR</span><span class="c10 c23 c2">M</span><span class="c10 c2">--</span><span class="c10 c2 c23">P</span><span class="c10 c2">LC...</span><span class="c2">&rsquo; by ignoring gaps. &nbsp;In addition, the scanner supports the use of multiple regular expressions simultaneously, which allows for the mapping of multiple motifs in a single scan. &nbsp;Using the extended ScanProsite syntax, simple strings can be rapidly searched. For example, &lsquo;</span><span class="c10 c2">MPSC</span><span class="c2">&rsquo; is the same as the full expression in ScanProsite syntax, &lsquo;</span><span class="c10 c2">M-P-S-C</span><span class="c2">&rsquo;.</span></p><p class="c3"><span class="c0 c12">Show GLD-1 SP and TP regular expression.</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c15 c14 c2">Directly Editing the Alignment</span></p><p class="c3"><span class="c2">In SlimShadey, an alignment is split into two panes. The left pane holds names of the sequences and names of the annotation rows. &nbsp;The right pane holds the alignment and annotations. Clicking on a sequence or annotation row&rsquo;s name, and left- or right-clicking somewhere on the alignment, will open a menu with options to manipulate properties of the alignment.</span></p><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">Sequence-level control (left pane, clicking on sequence name or annotation name)</span></p><p class="c3"><span class="c2">In an alignment in SlimShadey, the left pane</span><span class="c2">&nbsp;has the names of the annotations and sequences. Clicking on the name of a sequence or annotation allows the user to rename, reorder, or delete the selection.</span></p><ul class="c6 lst-kix_cbfpvidehrpt-0 start"><li class="c3 c8"><span class="c2 c5">Annotations: </span><span class="c0">Clicking on an annotation allows the user to edit the annotation row&rsquo;s name, add a new annotation row, change the relative position of an annotation row (up or down), and delete an annotation row.</span></li><li class="c3 c8"><span class="c2 c5">Sequence names: </span><span class="c0">Similar to annotation rows, sequences can be renamed, reordered, or deleted by clicking on the name. &nbsp;Sequences can also be reordered relative to a chosen sequence, based on pairwise scoring between every other sequence and the chosen sequence with a substitution matrix - the chosen sequence is moved to the top, and sequences are reordered below in order of highest- to lowest-scoring.</span></li><li class="c3 c8"><span class="c2 c5">Consensus: </span><span class="c2">If changes are made to the sequence alignment without realignment, the consensus can be recalculated by clicking on &ldquo;</span><span class="c10 c2">Consensus</span><span class="c2">&rdquo; in the left pane at the bottom of the sequence alignment. &nbsp;If the sequences are realigned, the consensus sequence is automatically recalculated based on simple majority.</span></li></ul><p class="c1"><span class="c4 c2"></span></p><p class="c3"><span class="c2 c4">Active functions (right pane, left-click or right-click on alignment)</span></p><p class="c3"><span class="c0">SlimShadey allows for residue level control of editing, shading, and recoloring of sequence alignments. &nbsp;These options can be accessed by right- or left-clicking directly on a character in the alignment. &nbsp;This includes the following functions:</span></p><ul class="c6 lst-kix_yoy1ormm4wf-0 start"><li class="c3 c8"><span class="c2 c5">Residue-level control:</span><span class="c0">&nbsp;editing (changing, deleting, inserting) &nbsp;individual residues and their properties (color, shade)</span></li><li class="c3 c8"><span class="c2 c5">Column-level control: </span><span class="c0">deleting individual columns or groups of columns, editing properties of groups of columns, moving subsequences to new tabs for downstream analysis, hading subsequences to indicate new motifs, domains, or regions</span></li></ul><p class="c3"><span class="c0">The active functions are implemented as follows:</span></p><ul class="c6 lst-kix_weixxh5ko91p-0 start"><li class="c3 c8"><span class="c14 c2">Left-click</span><span class="c0">&nbsp;on sequence alignment activates the residue level edit function. &nbsp;This allows for the following processes: &nbsp;</span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-1 start"><li class="c3 c7"><span class="c2 c5">Edit: </span><span class="c0">From the edit strip at the bottom of the window the properties of the character can be changed. &nbsp;These properties include the character identify, shading color, and text color can be modified. </span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-0"><li class="c3 c8"><span class="c14 c2">Right-click</span><span class="c0">&nbsp;on the sequence alignment activates the column editing function. &nbsp;This allows for the following processes:</span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-1 start"><li class="c3 c7"><span class="c2 c5">Insert left/right</span><span class="c0">: Allows for the insertion of a new column to the left or right of the position identified by the right-click (green shade column indicator).</span></li><li class="c3 c7"><span class="c2 c5">Delete selection:</span><span class="c2">&nbsp; This allows the d</span><span class="c0">eletion of individual columns or groups of columns defined with the right-click-and-drag functionality.</span></li><li class="c3 c7"><span class="c2 c5">Make sequence logo: </span><span class="c2">Sequences that are defined by </span><span class="c0">Generation of sequence logos from subsequences right-click-and-drag can be converted to sequence logo &nbsp;in PNG format that can be printed or inserted in presentations.</span></li><li class="c3 c7"><span class="c2 c5">Annotate selection: </span><span class="c2">Allows block annotation of selected subsequences (see above, Add annotation row</span><span class="c4 c2">)</span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-2 start"><li class="c3 c17"><span class="c2">If 5 columns are selected and the user enters the annotation &lsquo;</span><span class="c10 c2">A</span><span class="c2">&rsquo;, then the annotation produced will extend the pattern to &lsquo;</span><span class="c10 c2">AAAAA</span><span class="c2">&rsquo;. Similarly, if the user enters &lsquo;</span><span class="c10 c2">AB</span><span class="c2">&rsquo;, the annotation will be &lsquo;</span><span class="c10 c2">ABABA</span><span class="c2">&rsquo;. This allows quickly adding an annotation for multiple columns with a fixed sequence, e.g. &lsquo;</span><span class="c10 c2">&rArr;&rArr;...&rArr;&rArr;</span><span class="c2">&rsquo; by selecting those columns, </span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-1"><li class="c3 c7"><span class="c2 c5">Edit sequences</span><span class="c2">: This function allows for the c</span><span class="c0">olumn level editing of individual residues</span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-2 start"><li class="c3 c17"><span class="c2">Like the option above, a smaller subsequence can be entered and extended (e.g. the user can select 3 columns, enter &lsquo;</span><span class="c10 c2">A</span><span class="c2">&rsquo; for a sequence, and that sequence&rsquo;s characters will be &lsquo;</span><span class="c10 c2">AAA</span><span class="c0">&rsquo; in those columns). Alternatively, the user can choose to infer </span></li></ul><ul class="c6 lst-kix_weixxh5ko91p-1"><li class="c3 c7"><span class="c2 c5">Shade manually: </span><span class="c2">Manual shading provides c</span><span class="c0">ontrol of shading individual sequences</span></li><li class="c3 c7"><span class="c2 c5">Block shading: </span><span class="c0">Fully manual control of shading over motifs, domains, novel regions, and sections of annotations</span></li><li class="c3 c7"><span class="c2 c5">Export selection: </span><span class="c0">The user can select right-click-and-drag to identify the subsequences to export. &nbsp;The selected sequence, with the names retained, will be exported to a new tab that is independent from the original. &nbsp;New shading rules and additional daughter tabs can be generated as desired and saved independently.</span></li></ul><p class="c3"><span class="c14 c2">NOTE: </span><span class="c0">An editing warning is included to warn the user of unintended changes that may alter the information content in the alignment. &nbsp;The user can turn off the warning and free-edit the sequence alignment to make multiple changes in sequence. &nbsp;</span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><hr style="page-break-before:always;display:none;"><p class="c1"><span class="c15 c14 c2"></span></p><p class="c3"><span class="c15 c14 c2">Troubleshooting and FAQ</span></p><p class="c3"><span class="c9 c2 c5">What if SlimShadey fails to launch?</span></p><p class="c3"><span class="c0">Use the terminal and use the &ldquo;java -version&rdquo; command to check your version of Java JRE. &nbsp;SlimShadey requires Java 1.8.0 or higher.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">What if you have checked your Java version and SlimShady still does not launch?</span></p><p class="c3"><span class="c2">In rare cases, Java JRE may need to be uninstalled and reinstalled. &nbsp;Follow the instructions for uninstalling all Java components </span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.java.com/en/download/help/uninstall_java.xml&amp;sa=D&amp;ust=1603771834458000&amp;usg=AOvVaw2VvYuJ-9wFz9635us-5chG">here</a></span><span class="c2">, reboot your system, reinstall Java (</span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.oracle.com/java/technologies/javase-jre8-downloads.html&amp;sa=D&amp;ust=1603771834458000&amp;usg=AOvVaw2ze6IqrzoAg0R8RGmAe6Pc">Oracle</a></span><span class="c2">), and reboot your system again</span><span class="c0">.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">What if SlimShadey is running slowly?</span></p><p class="c3"><span class="c0">Slimshadey has been designed to run on legacy systems; however, it will run more slowly on systems that only support 32-bit versions of Java. &nbsp;Significant performance increases can be obtained by moving to hardware that can support 64-bit versions. &nbsp;The performance of SlimShadey can also be improved by using a system with at least 8GB of memory and a dedicated GPU. &nbsp;If larger datasets are being processed then SlimShadey will require additional memory (&gt;16GB) and faster CPU (e.g. Intel Core i7 or higher).</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2 c5">What if SlimShadey is running slowly on a system with updated hardware?</span></p><p class="c3"><span class="c0">Depending on your system configuration you may have to increase the memory available to SlimShadey for better performance. &nbsp;Using the terminal from the SlimShadey directory launch the software using the following command:</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c10 c2 c13">java -Xmx16g -Xms16g -jar SlimShadey.jar</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">16g is the memory allocated to the JVM gigabytes. &nbsp;This value can be increased to the limit of the memory available on your machine (e.g. 32g, 64g, etc.). &nbsp;If you have less than 8GB of memory then use the following:</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c13 c10 c2">java -Xmx2048g -Xms2048g -jar SlimShadey.jar</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">2048m is the memory allocated to the JVM in megabytes (or 2 gigabytes). &nbsp;This value can be increased to the limit of the memory available on your machine in multiples of 1024.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">How many tabs can I have open?</span></p><p class="c3"><span class="c0">We have not restricted the number of tabs that can be open at one time. &nbsp;Each tab is treated as a separate project but only the active tab is rendered in real-time. &nbsp;Depending on your system resources and memory allocation, SlimShadey may run more slowly if a large number of tabs are open. &nbsp;Since only the active tab is rendered in real time, having multiple tabs open does not impact the performance of SlimShadey unless system resources are limited.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">Can I open multiple instances of SlimShadey?</span></p><p class="c3"><span class="c0">Yes. &nbsp;It is possible to open multiple instances of SlimShadey, however, the additional rendering may compromise the performance of real-time functions. &nbsp;If you are working with multiple sequences, we recommend using different tabs in a single instance for the best performance.</span></p><p class="c1"><span class="c9 c2 c5"></span></p><p class="c3"><span class="c2 c5 c9">What if I have a large dataset?</span></p><p class="c3"><span class="c0">If you are working with a large dataset then you may need to increase the memory allocated to the JVM (Java Virtual Machine). &nbsp;See troubleshooting on how to get better performance on updated hardware (above). </span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2 c5">Is there a limit to the number of sequences SlimShadey can handle?</span></p><p class="c3"><span class="c0">Slimshadey was designed for the laboratory scale analysis of sequence data; however, we have not limited the number of sequences that can be processed. &nbsp;The upper limit of sequence that can be analyzed will be determined by your system hardware, memory available to the JVM, and the type of processing (visualization vs alignment). If you have a genome scale dataset then you will have to use a different software package. &nbsp;Based on the sample system used for this work (Intel Core i7 8750H, 16GB RAM, NVIDIA GTX 1060), 200 sequences with approximately 1000 characters each can be processed without significant lagging. &nbsp;Larger datasets or less robust hardware may result in some lag in the real-time processing functions.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2 c5">Do I have to use OpalFX to align my sequences?</span></p><p class="c3"><span class="c0">No. &nbsp;SlimShadey will accept any aligned or unaligned text file in FASTA format. &nbsp;We have provided Opal FX as an option if you want to align your sequence inside a SlimShadey session. &nbsp;This option is particularly useful if you export a subsequence to a new tab or delete a sequence from the original alignment and need to realign. &nbsp;Overall, the Opal algorithm has similar performance to MAFTT and Muscle (Wheeler, 2007).</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">What if the shading looks strange or unexpected?</span></p><p class="c3"><span class="c2">SlimShadey is optimized for shading, highlighting, and editing of </span><span class="c2 c5 c16">aligned</span><span class="c0">&nbsp;nucleic acids and protein sequences. &nbsp;You can align the sequences using the algorithm of your choice and import the aligned sequences into SlimShadey as a FASTA file or you can use the included Opal FX to align the sequences in the tab you are working in.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">What if the shading in my RTF file does not match the SlimShadey window?</span></p><p class="c3"><span class="c0">SlimShadey adheres strictly to the rules for the RTF file format. &nbsp;Software such as Microsoft Word may not render all color, shading, and characters correctly. To guarantee an accurate representation in Microsoft Word RTF, use grayscale and standard UTF-8 characters. Most colors are supported, though some may be omitted or converted to grayscale by Microsoft Word. We recommend saving to a PNG file for WYSIWYG output.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c9 c2 c5">When working with sequencing reads the collapse function does not work properly?</span></p><p class="c3"><span class="c0">FASTA files of sequencing reads can contain variable numbers of unknown (N or X) characters at the 5&rsquo; and 3&rsquo; ends. &nbsp;SlimShadey treats sequences with any differences as unique and will not collapse them into a single group. &nbsp;Option 1: The 5&rsquo; and 3&rsquo; ends can be trimmed using the active right-click drag function to delete the columns with unknown residues. &nbsp;Option 2: The active right-click drag function can also be used to select a range of residues and transfer them to a new tab for subsequent analysis. &nbsp;Either approach will allow for the maximum reduction (grouping) of the dataset with reads with unknown residues are eliminated.</span></p><p class="c3"><span class="c9 c12 c2 c5">Can I write my own shading rules?</span></p><p class="c3"><span class="c0">Yes. &nbsp;For the identification of novel motifs and domains, SlimShadey supports shading using ScanProsite regular expressions. If you would like to write your own rules for shading, the rule sets for are in the following classes and data files:</span></p><p class="c1"><span class="c0 c12"></span></p><p class="c3"><span class="c9 c2 c5">Can I modify classes and datafiles?</span></p><p class="c3"><span class="c0">Yes. &nbsp;SlimShadey is provided with all source code. &nbsp;Classes can be modified and recompiled to customize options or add functionality to SlimShadey. &nbsp;Slimshadey was developed using JDK 8 and the NetBeans IDE. &nbsp;If you would like to edit it in the same environment as it was developed then follow the do the following:</span></p><ul class="c6 lst-kix_ahj95ozgae0m-0 start"><li class="c3 c8"><span class="c0">Delete NetBeans if it has been previously installed</span></li><li class="c3 c8"><span class="c0">Delete all Java versions, JDKs, and updates. </span></li></ul><ul class="c6 lst-kix_ahj95ozgae0m-1 start"><li class="c3 c7"><span class="c0">NOTE: If you delete these first then Netbeans uninstaller for Windows 10 will not work. &nbsp;If you deleted them then you need to reinstall them and then delete Netbeans - then delete all Java versions</span></li></ul><ul class="c6 lst-kix_ahj95ozgae0m-0"><li class="c3 c8"><span class="c0">Restart</span></li><li class="c3 c8"><span class="c2">Install JDK 8u111 with NetBeans 8.2 from Oracle downloadable at this </span><span class="c16 c18 c2"><a class="c11" href="https://www.google.com/url?q=https://www.oracle.com/technetwork/java/javase/downloads/jdk-netbeans-jsp-3413139-esa.html&amp;sa=D&amp;ust=1603771834463000&amp;usg=AOvVaw2s4olMM8VNSj2T8DfOFy1U">link</a></span></li></ul><ul class="c6 lst-kix_ahj95ozgae0m-1 start"><li class="c3 c7"><span class="c0">JDK 8 and NetBeans 8.2 can be obtained separately; however, this installs them together so you don&rsquo;t have to set the paths later and the versions are known to be compatible</span></li></ul><p class="c1"><span class="c0"></span></p><hr style="page-break-before:always;display:none;"><p class="c1"><span class="c15 c14 c2"></span></p><p class="c3"><span class="c16 c14 c2">References</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Hulo, N., Sigrist, C.J., Le Saux, V., Langendijk&#8208;Genevaux, P.S., Bordoli, L., Gattiker, A., De Castro, E., Bucher, P. and Bairoch, A., (2004) Recent improvements to the PROSITE database. Nucleic Acids Research, 32(suppl_1), pp.D134-D137</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Jeanmougin F, Thompson JD, Gouy M, Higgins DG, Gibson TJ. Multiple sequence alignment with Clustal X. Trends Biochem Sci. 1998;23(10):403-405. doi:10.1016/s0968-0004(98)01285-7</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Larkin MA, Blackshields G, Brown NP, Chenna R, McGettigan PA, McWilliam H, Valentin F, Wallace IM, Wilm A, Lopez R, Thompson JD, Gibson TJ, Higgins DG.</span></p><p class="c3"><span class="c0">(2007). Clustal W and Clustal X version 2.0. Bioinformatics, 23, 2947-2948.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Lee MH, Schedl T. C. elegans star proteins, GLD-1 and ASD-2, regulate specific RNA targets to control development. Adv Exp Med Biol. 2010;693:106-122. doi:10.1007/978-1-4419-7005-3_8.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Madeira F, Park YM, Lee J, et al. The EMBL-EBI search and sequence analysis tools APIs in 2019. Nucleic Acids Research. 2019 Jul;47(W1):W636-W641. DOI: 10.1093/nar/gkz268.</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Potter SC, Luciani A, Eddy SR, Park Y, Lopez R, Finn RD. HMMER web server: 2018 update. Nucleic Acids Res. 2018;46(W1):W200-W204. doi:10.1093/nar/gky448</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Sayle RA, Milner-White EJ. RASMOL: biomolecular graphics for all. Trends Biochem Sci. 1995;20(9):374. doi:10.1016/s0968-0004(00)89080-5</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Senbanjo LT, Chellaiah MA. CD44: A Multifunctional Cell Surface Adhesion Receptor Is a Regulator of Progression and Metastasis of Cancer Cells. Front Cell Dev Biol. 2017;5:18. Published 2017 Mar 7. doi:10.3389/fcell.2017.00018</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Sigrist CJ, de Castro E, Cerutti L, et al. New and continuing developments at PROSITE. Nucleic Acids Res. 2013;41(Database issue):D344-D347. doi:10.1093/nar/gks1067</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c0">Thompson JD, Higgins DG, Gibson TJ. CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice. Nucleic Acids Res. 1994;22(22):4673-4680. doi:10.1093/nar/22.22.4673</span></p><p class="c1"><span class="c0"></span></p><p class="c3"><span class="c2">Wheeler TJ, Kececioglu JD. Multiple alignment by aligning alignments. Bioinformatics. 2007;23(13):i559-i568. doi:10.1093/bioinformatics/btm226</span></p><div><p class="c24 c28"><span class="c9 c27"></span></p></div></div></body></html>