<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ul.lst-kix_nh8twm5hcw9o-2{list-style-type:none}ul.lst-kix_s8ljebqdwq0-4{list-style-type:none}.lst-kix_1pwbqchre05h-2>li{counter-increment:lst-ctn-kix_1pwbqchre05h-2}.lst-kix_94bjxqinr4qs-6>li:before{content:"" counter(lst-ctn-kix_94bjxqinr4qs-6,decimal) ". "}ul.lst-kix_nh8twm5hcw9o-3{list-style-type:none}ul.lst-kix_s8ljebqdwq0-5{list-style-type:none}.lst-kix_reo1rr5lb8ia-3>li:before{content:"\0025cf  "}.lst-kix_reo1rr5lb8ia-4>li:before{content:"\0025cb  "}ul.lst-kix_nh8twm5hcw9o-0{list-style-type:none}ul.lst-kix_s8ljebqdwq0-2{list-style-type:none}.lst-kix_94bjxqinr4qs-5>li:before{content:"(" counter(lst-ctn-kix_94bjxqinr4qs-5,lower-roman) ") "}.lst-kix_94bjxqinr4qs-7>li:before{content:"" counter(lst-ctn-kix_94bjxqinr4qs-7,lower-latin) ". "}ul.lst-kix_nh8twm5hcw9o-1{list-style-type:none}ul.lst-kix_s8ljebqdwq0-3{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-6{list-style-type:none}ul.lst-kix_s8ljebqdwq0-0{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-7{list-style-type:none}ul.lst-kix_s8ljebqdwq0-1{list-style-type:none}.lst-kix_reo1rr5lb8ia-2>li:before{content:"\0025a0  "}.lst-kix_reo1rr5lb8ia-6>li:before{content:"\0025cf  "}ul.lst-kix_nh8twm5hcw9o-4{list-style-type:none}ul.lst-kix_nh8twm5hcw9o-5{list-style-type:none}ul.lst-kix_kiw2208yjgf-5{list-style-type:none}ul.lst-kix_kiw2208yjgf-4{list-style-type:none}.lst-kix_742seixkzi4g-1>li{counter-increment:lst-ctn-kix_742seixkzi4g-1}.lst-kix_reo1rr5lb8ia-0>li:before{content:"\0025cf  "}.lst-kix_reo1rr5lb8ia-7>li:before{content:"\0025cb  "}.lst-kix_reo1rr5lb8ia-8>li:before{content:"\0025a0  "}ul.lst-kix_kiw2208yjgf-7{list-style-type:none}.lst-kix_94bjxqinr4qs-3>li:before{content:"(" counter(lst-ctn-kix_94bjxqinr4qs-3,decimal) ") "}ul.lst-kix_kiw2208yjgf-6{list-style-type:none}ol.lst-kix_1pwbqchre05h-1.start{counter-reset:lst-ctn-kix_1pwbqchre05h-1 0}ul.lst-kix_kiw2208yjgf-1{list-style-type:none}.lst-kix_94bjxqinr4qs-4>li:before{content:"(" counter(lst-ctn-kix_94bjxqinr4qs-4,lower-latin) ") "}ul.lst-kix_kiw2208yjgf-0{list-style-type:none}.lst-kix_reo1rr5lb8ia-1>li:before{content:"\0025cb  "}ul.lst-kix_kiw2208yjgf-3{list-style-type:none}ul.lst-kix_kiw2208yjgf-2{list-style-type:none}ol.lst-kix_94bjxqinr4qs-8.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-8 0}ol.lst-kix_742seixkzi4g-8.start{counter-reset:lst-ctn-kix_742seixkzi4g-8 0}.lst-kix_94bjxqinr4qs-2>li:before{content:"" counter(lst-ctn-kix_94bjxqinr4qs-2,lower-roman) ") "}ul.lst-kix_nh8twm5hcw9o-8{list-style-type:none}.lst-kix_94bjxqinr4qs-1>li:before{content:"" counter(lst-ctn-kix_94bjxqinr4qs-1,lower-latin) ") "}ul.lst-kix_s8ljebqdwq0-8{list-style-type:none}.lst-kix_94bjxqinr4qs-0>li:before{content:"" counter(lst-ctn-kix_94bjxqinr4qs-0,decimal) ") "}.lst-kix_reo1rr5lb8ia-5>li:before{content:"\0025a0  "}ul.lst-kix_s8ljebqdwq0-6{list-style-type:none}ul.lst-kix_s8ljebqdwq0-7{list-style-type:none}ol.lst-kix_94bjxqinr4qs-0{list-style-type:none}ol.lst-kix_94bjxqinr4qs-1{list-style-type:none}.lst-kix_94bjxqinr4qs-8>li:before{content:"" counter(lst-ctn-kix_94bjxqinr4qs-8,lower-roman) ". "}.lst-kix_s8ljebqdwq0-5>li:before{content:"\0025a0  "}.lst-kix_vknp6nujnm9e-3>li:before{content:"\0025cf  "}.lst-kix_vknp6nujnm9e-5>li:before{content:"\0025a0  "}.lst-kix_s8ljebqdwq0-2>li:before{content:"\0025a0  "}.lst-kix_s8ljebqdwq0-6>li:before{content:"\0025cf  "}.lst-kix_vknp6nujnm9e-4>li:before{content:"\0025cb  "}.lst-kix_s8ljebqdwq0-1>li:before{content:"\0025cb  "}.lst-kix_vknp6nujnm9e-7>li:before{content:"\0025cb  "}.lst-kix_s8ljebqdwq0-7>li:before{content:"\0025cb  "}.lst-kix_s8ljebqdwq0-0>li:before{content:"\0025cf  "}.lst-kix_s8ljebqdwq0-8>li:before{content:"\0025a0  "}.lst-kix_vknp6nujnm9e-6>li:before{content:"\0025cf  "}ol.lst-kix_1pwbqchre05h-6.start{counter-reset:lst-ctn-kix_1pwbqchre05h-6 0}ol.lst-kix_94bjxqinr4qs-8{list-style-type:none}ol.lst-kix_94bjxqinr4qs-6{list-style-type:none}ol.lst-kix_94bjxqinr4qs-7{list-style-type:none}ol.lst-kix_94bjxqinr4qs-4{list-style-type:none}ol.lst-kix_94bjxqinr4qs-5{list-style-type:none}.lst-kix_vknp6nujnm9e-8>li:before{content:"\0025a0  "}ol.lst-kix_94bjxqinr4qs-2{list-style-type:none}ol.lst-kix_94bjxqinr4qs-3{list-style-type:none}.lst-kix_s8ljebqdwq0-3>li:before{content:"\0025cf  "}.lst-kix_s8ljebqdwq0-4>li:before{content:"\0025cb  "}ol.lst-kix_94bjxqinr4qs-3.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-3 0}ul.lst-kix_vknp6nujnm9e-8{list-style-type:none}.lst-kix_742seixkzi4g-8>li{counter-increment:lst-ctn-kix_742seixkzi4g-8}ul.lst-kix_vknp6nujnm9e-6{list-style-type:none}ul.lst-kix_vknp6nujnm9e-7{list-style-type:none}ol.lst-kix_1pwbqchre05h-0.start{counter-reset:lst-ctn-kix_1pwbqchre05h-0 0}ul.lst-kix_vknp6nujnm9e-4{list-style-type:none}ul.lst-kix_vknp6nujnm9e-5{list-style-type:none}ul.lst-kix_vknp6nujnm9e-2{list-style-type:none}ol.lst-kix_1pwbqchre05h-7.start{counter-reset:lst-ctn-kix_1pwbqchre05h-7 0}.lst-kix_742seixkzi4g-5>li{counter-increment:lst-ctn-kix_742seixkzi4g-5}ul.lst-kix_vknp6nujnm9e-3{list-style-type:none}.lst-kix_wulsapsxo6ef-7>li:before{content:"\0025cb  "}.lst-kix_wulsapsxo6ef-8>li:before{content:"\0025a0  "}.lst-kix_1pwbqchre05h-4>li{counter-increment:lst-ctn-kix_1pwbqchre05h-4}ol.lst-kix_94bjxqinr4qs-2.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-2 0}.lst-kix_wulsapsxo6ef-2>li:before{content:"\0025a0  "}.lst-kix_94bjxqinr4qs-3>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-3}.lst-kix_wulsapsxo6ef-1>li:before{content:"\0025cb  "}.lst-kix_wulsapsxo6ef-3>li:before{content:"\0025cf  "}ul.lst-kix_vknp6nujnm9e-0{list-style-type:none}.lst-kix_wulsapsxo6ef-6>li:before{content:"\0025cf  "}ul.lst-kix_vknp6nujnm9e-1{list-style-type:none}.lst-kix_wulsapsxo6ef-5>li:before{content:"\0025a0  "}.lst-kix_wulsapsxo6ef-4>li:before{content:"\0025cb  "}ol.lst-kix_742seixkzi4g-3{list-style-type:none}.lst-kix_nh8twm5hcw9o-4>li:before{content:"\0025cb  "}ol.lst-kix_742seixkzi4g-2{list-style-type:none}.lst-kix_weixxh5ko91p-0>li:before{content:"\0025cf  "}ol.lst-kix_742seixkzi4g-1{list-style-type:none}ol.lst-kix_742seixkzi4g-0{list-style-type:none}ol.lst-kix_742seixkzi4g-7{list-style-type:none}ol.lst-kix_742seixkzi4g-6{list-style-type:none}.lst-kix_742seixkzi4g-0>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-0,decimal) ". "}.lst-kix_742seixkzi4g-4>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-4,lower-latin) ". "}ol.lst-kix_742seixkzi4g-5{list-style-type:none}.lst-kix_cbfpvidehrpt-3>li:before{content:"\0025cf  "}.lst-kix_cbfpvidehrpt-7>li:before{content:"\0025cb  "}ol.lst-kix_742seixkzi4g-4{list-style-type:none}.lst-kix_nh8twm5hcw9o-0>li:before{content:"\0025cf  "}.lst-kix_nh8twm5hcw9o-8>li:before{content:"\0025a0  "}ul.lst-kix_reo1rr5lb8ia-0{list-style-type:none}.lst-kix_d0pnlbhmephp-6>li:before{content:"\0025cf  "}ul.lst-kix_reo1rr5lb8ia-1{list-style-type:none}.lst-kix_weixxh5ko91p-4>li:before{content:"\0025cb  "}ol.lst-kix_742seixkzi4g-8{list-style-type:none}ul.lst-kix_reo1rr5lb8ia-2{list-style-type:none}.lst-kix_742seixkzi4g-6>li{counter-increment:lst-ctn-kix_742seixkzi4g-6}ul.lst-kix_reo1rr5lb8ia-3{list-style-type:none}.lst-kix_nh8twm5hcw9o-2>li:before{content:"\0025a0  "}ul.lst-kix_reo1rr5lb8ia-4{list-style-type:none}.lst-kix_742seixkzi4g-2>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-2,lower-roman) ". "}ul.lst-kix_reo1rr5lb8ia-5{list-style-type:none}.lst-kix_cbfpvidehrpt-1>li:before{content:"\0025cb  "}ul.lst-kix_reo1rr5lb8ia-6{list-style-type:none}ul.lst-kix_hefi1acv3kbb-0{list-style-type:none}ul.lst-kix_hefi1acv3kbb-1{list-style-type:none}.lst-kix_weixxh5ko91p-6>li:before{content:"\0025cf  "}.lst-kix_weixxh5ko91p-8>li:before{content:"\0025a0  "}ul.lst-kix_hefi1acv3kbb-4{list-style-type:none}ul.lst-kix_hefi1acv3kbb-5{list-style-type:none}ul.lst-kix_hefi1acv3kbb-2{list-style-type:none}.lst-kix_742seixkzi4g-8>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-8,lower-roman) ". "}ul.lst-kix_hefi1acv3kbb-3{list-style-type:none}.lst-kix_d0pnlbhmephp-8>li:before{content:"\0025a0  "}.lst-kix_kiw2208yjgf-4>li:before{content:"\0025cb  "}.lst-kix_742seixkzi4g-6>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-6,decimal) ". "}.lst-kix_742seixkzi4g-7>li{counter-increment:lst-ctn-kix_742seixkzi4g-7}.lst-kix_kiw2208yjgf-2>li:before{content:"\0025a0  "}ul.lst-kix_90sjlybkt4o7-8{list-style-type:none}ul.lst-kix_90sjlybkt4o7-7{list-style-type:none}ul.lst-kix_90sjlybkt4o7-6{list-style-type:none}ul.lst-kix_90sjlybkt4o7-5{list-style-type:none}.lst-kix_rddsjt44cy73-4>li:before{content:"\0025cb  "}.lst-kix_kiw2208yjgf-0>li:before{content:"\0025cf  "}ol.lst-kix_94bjxqinr4qs-1.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-1 0}.lst-kix_1pwbqchre05h-5>li:before{content:"(" counter(lst-ctn-kix_1pwbqchre05h-5,lower-roman) ") "}.lst-kix_d0pnlbhmephp-2>li:before{content:"\0025a0  "}.lst-kix_d0pnlbhmephp-4>li:before{content:"\0025cb  "}.lst-kix_1pwbqchre05h-3>li:before{content:"(" counter(lst-ctn-kix_1pwbqchre05h-3,decimal) ") "}.lst-kix_1pwbqchre05h-7>li:before{content:"" counter(lst-ctn-kix_1pwbqchre05h-7,lower-latin) ". "}ul.lst-kix_90sjlybkt4o7-4{list-style-type:none}.lst-kix_rddsjt44cy73-2>li:before{content:"\0025a0  "}ul.lst-kix_90sjlybkt4o7-3{list-style-type:none}ul.lst-kix_90sjlybkt4o7-2{list-style-type:none}.lst-kix_d0pnlbhmephp-0>li:before{content:"\0025cf  "}ul.lst-kix_90sjlybkt4o7-1{list-style-type:none}.lst-kix_vknp6nujnm9e-1>li:before{content:"\0025cb  "}ul.lst-kix_90sjlybkt4o7-0{list-style-type:none}.lst-kix_rddsjt44cy73-0>li:before{content:"\0025cf  "}.lst-kix_nh8twm5hcw9o-6>li:before{content:"\0025cf  "}ol.lst-kix_1pwbqchre05h-8.start{counter-reset:lst-ctn-kix_1pwbqchre05h-8 0}.lst-kix_1pwbqchre05h-8>li{counter-increment:lst-ctn-kix_1pwbqchre05h-8}ol.lst-kix_742seixkzi4g-7.start{counter-reset:lst-ctn-kix_742seixkzi4g-7 0}.lst-kix_cbfpvidehrpt-5>li:before{content:"\0025a0  "}.lst-kix_1pwbqchre05h-1>li:before{content:"" counter(lst-ctn-kix_1pwbqchre05h-1,lower-latin) ") "}ol.lst-kix_94bjxqinr4qs-0.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-0 0}ul.lst-kix_rddsjt44cy73-2{list-style-type:none}ul.lst-kix_rddsjt44cy73-3{list-style-type:none}ul.lst-kix_rddsjt44cy73-0{list-style-type:none}ul.lst-kix_rddsjt44cy73-1{list-style-type:none}.lst-kix_rddsjt44cy73-6>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-0>li:before{content:"\0025cf  "}.lst-kix_yoy1ormm4wf-2>li:before{content:"\0025a0  "}.lst-kix_yoy1ormm4wf-0>li:before{content:"\0025cf  "}.lst-kix_rddsjt44cy73-8>li:before{content:"\0025a0  "}.lst-kix_hefi1acv3kbb-6>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-4>li:before{content:"\0025cb  "}.lst-kix_hefi1acv3kbb-8>li:before{content:"\0025a0  "}.lst-kix_94bjxqinr4qs-5>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-5}ul.lst-kix_hefi1acv3kbb-8{list-style-type:none}.lst-kix_kiw2208yjgf-6>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-2>li:before{content:"\0025a0  "}ul.lst-kix_hefi1acv3kbb-6{list-style-type:none}ul.lst-kix_hefi1acv3kbb-7{list-style-type:none}.lst-kix_kiw2208yjgf-8>li:before{content:"\0025a0  "}ul.lst-kix_reo1rr5lb8ia-7{list-style-type:none}ul.lst-kix_reo1rr5lb8ia-8{list-style-type:none}.lst-kix_yoy1ormm4wf-4>li:before{content:"\0025cb  "}ul.lst-kix_kiw2208yjgf-8{list-style-type:none}.lst-kix_yoy1ormm4wf-6>li:before{content:"\0025cf  "}.lst-kix_yoy1ormm4wf-8>li:before{content:"\0025a0  "}.lst-kix_weixxh5ko91p-2>li:before{content:"\0025a0  "}ul.lst-kix_ahj95ozgae0m-0{list-style-type:none}ol.lst-kix_94bjxqinr4qs-5.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-5 0}ul.lst-kix_ahj95ozgae0m-2{list-style-type:none}ul.lst-kix_ahj95ozgae0m-1{list-style-type:none}.lst-kix_94bjxqinr4qs-4>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-4}.lst-kix_742seixkzi4g-0>li{counter-increment:lst-ctn-kix_742seixkzi4g-0}.lst-kix_1pwbqchre05h-3>li{counter-increment:lst-ctn-kix_1pwbqchre05h-3}ol.lst-kix_1pwbqchre05h-4.start{counter-reset:lst-ctn-kix_1pwbqchre05h-4 0}ol.lst-kix_742seixkzi4g-0.start{counter-reset:lst-ctn-kix_742seixkzi4g-0 0}ul.lst-kix_rddsjt44cy73-6{list-style-type:none}ul.lst-kix_rddsjt44cy73-7{list-style-type:none}ul.lst-kix_rddsjt44cy73-4{list-style-type:none}ul.lst-kix_rddsjt44cy73-5{list-style-type:none}ul.lst-kix_rddsjt44cy73-8{list-style-type:none}.lst-kix_90sjlybkt4o7-8>li:before{content:"\0025a0  "}.lst-kix_90sjlybkt4o7-7>li:before{content:"\0025cb  "}ul.lst-kix_yoy1ormm4wf-8{list-style-type:none}.lst-kix_u5mt6v244o5o-5>li:before{content:"\0025a0  "}.lst-kix_u5mt6v244o5o-7>li:before{content:"\0025cb  "}ul.lst-kix_yoy1ormm4wf-7{list-style-type:none}.lst-kix_u5mt6v244o5o-6>li:before{content:"\0025cf  "}.lst-kix_90sjlybkt4o7-3>li:before{content:"\0025cf  "}.lst-kix_u5mt6v244o5o-1>li:before{content:"\0025cb  "}.lst-kix_90sjlybkt4o7-2>li:before{content:"\0025a0  "}.lst-kix_u5mt6v244o5o-0>li:before{content:"\0025cf  "}.lst-kix_u5mt6v244o5o-8>li:before{content:"\0025a0  "}.lst-kix_90sjlybkt4o7-0>li:before{content:"\0025cf  "}.lst-kix_94bjxqinr4qs-6>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-6}.lst-kix_90sjlybkt4o7-1>li:before{content:"\0025cb  "}.lst-kix_u5mt6v244o5o-2>li:before{content:"\0025a0  "}ol.lst-kix_1pwbqchre05h-8{list-style-type:none}ol.lst-kix_742seixkzi4g-6.start{counter-reset:lst-ctn-kix_742seixkzi4g-6 0}.lst-kix_u5mt6v244o5o-3>li:before{content:"\0025cf  "}ol.lst-kix_1pwbqchre05h-7{list-style-type:none}ol.lst-kix_1pwbqchre05h-6{list-style-type:none}ol.lst-kix_1pwbqchre05h-5{list-style-type:none}ol.lst-kix_1pwbqchre05h-4{list-style-type:none}.lst-kix_u5mt6v244o5o-4>li:before{content:"\0025cb  "}ol.lst-kix_1pwbqchre05h-3{list-style-type:none}.lst-kix_ahj95ozgae0m-4>li:before{content:"\0025cb  "}ol.lst-kix_1pwbqchre05h-2{list-style-type:none}ul.lst-kix_cbfpvidehrpt-3{list-style-type:none}ol.lst-kix_1pwbqchre05h-1{list-style-type:none}ul.lst-kix_cbfpvidehrpt-2{list-style-type:none}ol.lst-kix_1pwbqchre05h-0{list-style-type:none}ul.lst-kix_cbfpvidehrpt-5{list-style-type:none}ol.lst-kix_742seixkzi4g-5.start{counter-reset:lst-ctn-kix_742seixkzi4g-5 0}ul.lst-kix_cbfpvidehrpt-4{list-style-type:none}.lst-kix_ahj95ozgae0m-1>li:before{content:"\0025cb  "}.lst-kix_ahj95ozgae0m-5>li:before{content:"\0025a0  "}ul.lst-kix_cbfpvidehrpt-1{list-style-type:none}ul.lst-kix_cbfpvidehrpt-0{list-style-type:none}.lst-kix_ahj95ozgae0m-0>li:before{content:"\0025cf  "}.lst-kix_ahj95ozgae0m-8>li:before{content:"\0025a0  "}.lst-kix_1pwbqchre05h-1>li{counter-increment:lst-ctn-kix_1pwbqchre05h-1}.lst-kix_ahj95ozgae0m-6>li:before{content:"\0025cf  "}.lst-kix_ahj95ozgae0m-7>li:before{content:"\0025cb  "}ul.lst-kix_wulsapsxo6ef-2{list-style-type:none}ul.lst-kix_wulsapsxo6ef-3{list-style-type:none}ul.lst-kix_wulsapsxo6ef-0{list-style-type:none}ul.lst-kix_wulsapsxo6ef-1{list-style-type:none}ul.lst-kix_wulsapsxo6ef-6{list-style-type:none}ul.lst-kix_wulsapsxo6ef-7{list-style-type:none}.lst-kix_94bjxqinr4qs-0>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-0}ul.lst-kix_wulsapsxo6ef-4{list-style-type:none}ul.lst-kix_wulsapsxo6ef-5{list-style-type:none}ul.lst-kix_u5mt6v244o5o-7{list-style-type:none}.lst-kix_90sjlybkt4o7-4>li:before{content:"\0025cb  "}ul.lst-kix_ahj95ozgae0m-8{list-style-type:none}ul.lst-kix_yoy1ormm4wf-2{list-style-type:none}ul.lst-kix_u5mt6v244o5o-8{list-style-type:none}ul.lst-kix_ahj95ozgae0m-7{list-style-type:none}ul.lst-kix_yoy1ormm4wf-1{list-style-type:none}.lst-kix_90sjlybkt4o7-5>li:before{content:"\0025a0  "}ul.lst-kix_wulsapsxo6ef-8{list-style-type:none}ul.lst-kix_yoy1ormm4wf-0{list-style-type:none}ul.lst-kix_u5mt6v244o5o-3{list-style-type:none}.lst-kix_90sjlybkt4o7-6>li:before{content:"\0025cf  "}.lst-kix_ahj95ozgae0m-2>li:before{content:"\0025a0  "}ul.lst-kix_ahj95ozgae0m-4{list-style-type:none}ul.lst-kix_yoy1ormm4wf-6{list-style-type:none}ul.lst-kix_cbfpvidehrpt-7{list-style-type:none}ul.lst-kix_u5mt6v244o5o-4{list-style-type:none}ul.lst-kix_ahj95ozgae0m-3{list-style-type:none}.lst-kix_742seixkzi4g-2>li{counter-increment:lst-ctn-kix_742seixkzi4g-2}ul.lst-kix_yoy1ormm4wf-5{list-style-type:none}ul.lst-kix_cbfpvidehrpt-6{list-style-type:none}ul.lst-kix_u5mt6v244o5o-5{list-style-type:none}.lst-kix_ahj95ozgae0m-3>li:before{content:"\0025cf  "}ul.lst-kix_ahj95ozgae0m-6{list-style-type:none}ul.lst-kix_yoy1ormm4wf-4{list-style-type:none}ul.lst-kix_u5mt6v244o5o-6{list-style-type:none}ul.lst-kix_ahj95ozgae0m-5{list-style-type:none}ul.lst-kix_yoy1ormm4wf-3{list-style-type:none}ul.lst-kix_cbfpvidehrpt-8{list-style-type:none}.lst-kix_742seixkzi4g-3>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-3,decimal) ". "}.lst-kix_cbfpvidehrpt-6>li:before{content:"\0025cf  "}ul.lst-kix_u5mt6v244o5o-0{list-style-type:none}.lst-kix_nh8twm5hcw9o-5>li:before{content:"\0025a0  "}ul.lst-kix_u5mt6v244o5o-1{list-style-type:none}ul.lst-kix_d0pnlbhmephp-0{list-style-type:none}ul.lst-kix_u5mt6v244o5o-2{list-style-type:none}.lst-kix_cbfpvidehrpt-4>li:before{content:"\0025cb  "}.lst-kix_cbfpvidehrpt-8>li:before{content:"\0025a0  "}.lst-kix_nh8twm5hcw9o-3>li:before{content:"\0025cf  "}.lst-kix_nh8twm5hcw9o-7>li:before{content:"\0025cb  "}ul.lst-kix_d0pnlbhmephp-6{list-style-type:none}.lst-kix_d0pnlbhmephp-7>li:before{content:"\0025cb  "}.lst-kix_cbfpvidehrpt-2>li:before{content:"\0025a0  "}.lst-kix_nh8twm5hcw9o-1>li:before{content:"\0025cb  "}ul.lst-kix_d0pnlbhmephp-5{list-style-type:none}ul.lst-kix_d0pnlbhmephp-8{list-style-type:none}ol.lst-kix_742seixkzi4g-4.start{counter-reset:lst-ctn-kix_742seixkzi4g-4 0}ul.lst-kix_d0pnlbhmephp-7{list-style-type:none}.lst-kix_wulsapsxo6ef-0>li:before{content:"\0025cf  "}.lst-kix_742seixkzi4g-1>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-1,lower-latin) ". "}ul.lst-kix_d0pnlbhmephp-2{list-style-type:none}.lst-kix_d0pnlbhmephp-5>li:before{content:"\0025a0  "}.lst-kix_weixxh5ko91p-5>li:before{content:"\0025a0  "}ul.lst-kix_d0pnlbhmephp-1{list-style-type:none}ul.lst-kix_d0pnlbhmephp-4{list-style-type:none}ul.lst-kix_d0pnlbhmephp-3{list-style-type:none}.lst-kix_weixxh5ko91p-7>li:before{content:"\0025cb  "}ol.lst-kix_742seixkzi4g-1.start{counter-reset:lst-ctn-kix_742seixkzi4g-1 0}.lst-kix_cbfpvidehrpt-0>li:before{content:"\0025cf  "}.lst-kix_1pwbqchre05h-7>li{counter-increment:lst-ctn-kix_1pwbqchre05h-7}.lst-kix_742seixkzi4g-7>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-7,lower-latin) ". "}.lst-kix_742seixkzi4g-5>li:before{content:"" counter(lst-ctn-kix_742seixkzi4g-5,lower-roman) ". "}ul.lst-kix_weixxh5ko91p-8{list-style-type:none}.lst-kix_kiw2208yjgf-3>li:before{content:"\0025cf  "}ul.lst-kix_weixxh5ko91p-7{list-style-type:none}ul.lst-kix_weixxh5ko91p-6{list-style-type:none}ul.lst-kix_weixxh5ko91p-5{list-style-type:none}ol.lst-kix_1pwbqchre05h-5.start{counter-reset:lst-ctn-kix_1pwbqchre05h-5 0}.lst-kix_kiw2208yjgf-1>li:before{content:"\0025cb  "}ol.lst-kix_94bjxqinr4qs-4.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-4 0}.lst-kix_rddsjt44cy73-3>li:before{content:"\0025cf  "}ol.lst-kix_1pwbqchre05h-2.start{counter-reset:lst-ctn-kix_1pwbqchre05h-2 0}.lst-kix_1pwbqchre05h-8>li:before{content:"" counter(lst-ctn-kix_1pwbqchre05h-8,lower-roman) ". "}.lst-kix_d0pnlbhmephp-3>li:before{content:"\0025cf  "}.lst-kix_1pwbqchre05h-6>li:before{content:"" counter(lst-ctn-kix_1pwbqchre05h-6,decimal) ". "}ul.lst-kix_weixxh5ko91p-0{list-style-type:none}.lst-kix_94bjxqinr4qs-7>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-7}ol.lst-kix_94bjxqinr4qs-7.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-7 0}.lst-kix_rddsjt44cy73-1>li:before{content:"\0025cb  "}.lst-kix_d0pnlbhmephp-1>li:before{content:"\0025cb  "}ul.lst-kix_weixxh5ko91p-4{list-style-type:none}.lst-kix_1pwbqchre05h-4>li:before{content:"(" counter(lst-ctn-kix_1pwbqchre05h-4,lower-latin) ") "}ul.lst-kix_weixxh5ko91p-3{list-style-type:none}ul.lst-kix_weixxh5ko91p-2{list-style-type:none}.lst-kix_vknp6nujnm9e-2>li:before{content:"\0025a0  "}ul.lst-kix_weixxh5ko91p-1{list-style-type:none}.lst-kix_1pwbqchre05h-5>li{counter-increment:lst-ctn-kix_1pwbqchre05h-5}.lst-kix_94bjxqinr4qs-1>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-1}.lst-kix_vknp6nujnm9e-0>li:before{content:"\0025cf  "}.lst-kix_1pwbqchre05h-2>li:before{content:"" counter(lst-ctn-kix_1pwbqchre05h-2,lower-roman) ") "}.lst-kix_1pwbqchre05h-0>li:before{content:"" counter(lst-ctn-kix_1pwbqchre05h-0,decimal) ") "}.lst-kix_742seixkzi4g-3>li{counter-increment:lst-ctn-kix_742seixkzi4g-3}.lst-kix_yoy1ormm4wf-3>li:before{content:"\0025cf  "}.lst-kix_rddsjt44cy73-5>li:before{content:"\0025a0  "}.lst-kix_rddsjt44cy73-7>li:before{content:"\0025cb  "}.lst-kix_yoy1ormm4wf-1>li:before{content:"\0025cb  "}.lst-kix_742seixkzi4g-4>li{counter-increment:lst-ctn-kix_742seixkzi4g-4}ol.lst-kix_742seixkzi4g-3.start{counter-reset:lst-ctn-kix_742seixkzi4g-3 0}.lst-kix_1pwbqchre05h-0>li{counter-increment:lst-ctn-kix_1pwbqchre05h-0}.lst-kix_94bjxqinr4qs-2>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-2}.lst-kix_hefi1acv3kbb-7>li:before{content:"\0025cb  "}.lst-kix_kiw2208yjgf-5>li:before{content:"\0025a0  "}ol.lst-kix_94bjxqinr4qs-6.start{counter-reset:lst-ctn-kix_94bjxqinr4qs-6 0}.lst-kix_94bjxqinr4qs-8>li{counter-increment:lst-ctn-kix_94bjxqinr4qs-8}.lst-kix_kiw2208yjgf-7>li:before{content:"\0025cb  "}.lst-kix_hefi1acv3kbb-3>li:before{content:"\0025cf  "}.lst-kix_1pwbqchre05h-6>li{counter-increment:lst-ctn-kix_1pwbqchre05h-6}.lst-kix_hefi1acv3kbb-1>li:before{content:"\0025cb  "}li.li-bullet-0:before{margin-left:-18pt;white-space:nowrap;display:inline-block;min-width:18pt}ol.lst-kix_1pwbqchre05h-3.start{counter-reset:lst-ctn-kix_1pwbqchre05h-3 0}ol.lst-kix_742seixkzi4g-2.start{counter-reset:lst-ctn-kix_742seixkzi4g-2 0}.lst-kix_yoy1ormm4wf-5>li:before{content:"\0025a0  "}.lst-kix_yoy1ormm4wf-7>li:before{content:"\0025cb  "}.lst-kix_weixxh5ko91p-3>li:before{content:"\0025cf  "}.lst-kix_hefi1acv3kbb-5>li:before{content:"\0025a0  "}.lst-kix_weixxh5ko91p-1>li:before{content:"\0025cb  "}ol{margin:0;padding:0}table td,table th{padding:0}.c1{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Arial";font-style:normal}.c3{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c0{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:center}.c7{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Courier New";font-style:normal}.c20{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:right}.c4{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-family:"Arial";font-style:normal}.c19{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-family:"Arial"}.c21{padding-top:0pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c11{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c16{color:#000000;vertical-align:baseline;font-family:"Arial";font-style:normal}.c30{background-color:#ffffff;max-width:511.2pt;padding:50.4pt 50.4pt 50.4pt 50.4pt}.c31{color:#000000;text-decoration:none;vertical-align:baseline}.c13{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;text-decoration:underline}.c29{border:1px solid black;margin:5px}.c23{margin-left:108pt;padding-left:0pt}.c12{margin-left:72pt;padding-left:0pt}.c14{font-family:"Courier New";font-weight:400}.c10{padding:0;margin:0}.c6{color:inherit;text-decoration:inherit}.c8{margin-left:36pt;padding-left:0pt}.c26{font-style:normal}.c2{font-size:12pt}.c27{text-indent:36pt}.c9{font-weight:700}.c17{font-family:"Courier New"}.c28{margin-left:36pt}.c5{height:11pt}.c15{background-color:#ffff00}.c18{font-style:italic}.c24{font-weight:400}.c25{font-size:11pt}.c22{background-color:#cccccc}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c30"><p class="c3 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 367.00px; height: 153.20px;"><img alt="" src="images/image5.png" style="width: 418.57px; height: 297.35px; margin-left: -26.54px; margin-top: -16.80px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0 c5"><span class="c4 c2"></span></p><p class="c0"><span class="c4 c2">SlimShadey: JavaFX based creation, visualization, and editing of multiple sequence alignments</span></p><p class="c0"><span class="c2 c9">User </span><span class="c4 c2">Manual</span></p><p class="c0"><span class="c1">Avi Shah and Sudhir Nayak</span></p><p class="c0"><span class="c2">The College of New Jersey</span><hr style="page-break-before:always;display:none;"></p><p class="c3"><span class="c16 c13 c2 c9">Contents</span></p><p class="c3"><span class="c11 c2 c9"><a class="c6" href="#id.ctps4mkdca7y">Introduction</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.ltgos44kk4i8">What is SlimShadey?</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.lxv2blovq986">Requirements</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.1clwclydrbfq">Quick start video tutorial</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.f9y332j9rvnj">Performance</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.yn88mcbs0yxp">Checks to run before starting</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.tzt37jgbtgkx">Input file formats</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.xge2wndr30p1">Supported formats</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.l8ltlbp36y9h">How to input files to SlimShadey</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.eki70i1pyyw9">SlimShadey format (.mm)</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.sd310g36q790">Sample data provided</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.6tj099fx7d07">Parameter file</a></span></p><p class="c3"><span class="c11 c2 c9"><a class="c6" href="#id.hm2xctozcu7z">File menu options</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.alvel8j40h1v">Realign sequences (align sequences)</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.e6t9n9f2hum4">Print preview</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.dvl2g4twj8lk">Output to RTF (rich text format)</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.d6fymjflnuw">Save alignment as text</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.hpjwed9l5clo">Save as project</a></span></p><p class="c3"><span class="c11 c2 c9"><a class="c6" href="#id.fj721mwizur4">Edit menu options</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.bqbga6li8nbt">Session settings</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.grx1d1qdwxnj">First index</a></span><span class="c4 c2">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p><p class="c3 c27 c28"><span class="c11 c2 c9"><a class="c6" href="#id.9yvme81q8cbt">Font size</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.av7fmqwu0if0">Luminance threshold</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.xnfioacy2ouj">Hover refresh rate</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.tpunrwq89nsk">Gap only columns</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.a3ggyvgw0br">Calculate consensus</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.9ve34sqice01">Add annotation row</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.vlmi6y94qjfg">Collapse identical sequences</a></span></p><p class="c3"><span class="c11 c2 c9"><a class="c6" href="#id.9zckf9ky6mjt">Shade menu options</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.b9onwk15hiku">Shade by unique residues</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.9gn34rpv1t23">Shade by residue frequency</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.fmgkz650ryq6">Shade by sequence</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.gvklkfo29x3u">Shade with substitution matrix</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.t7fqmdtg27dm">Custom Prosite regex shading</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.4i3na39zeuwm">Prosite-based database shading</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.g8bqwfhft4s6">hmmscan based shading</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.qa7xocpx59fi">Default RASMOL coloring</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.6vr4ddq4bwzs">Default ClustalX coloring</a></span></p><p class="c3"><span class="c11 c2 c9"><a class="c6" href="#id.vdf66cgu6ys3">Directly editing the alignment</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.8ya7dcerna4n">Sequence-level control</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.48n4ln4lu5z3">Annotations</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.jor39ylau6a2">Sequence names</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.aoyjfs1u5b2e">Consensus</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.q0a5d6dxwhgo">Active functions</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.dm56sspqkdti">Residue-level control</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.ajyykno5oph5">Column-level control</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.j0ntpg7wy8e7">Left-click</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.4hsb458l6kzn">Edit</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.h3u7kqqh0hxy">Right-click</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.y1pi0a97bnyw">Insert left/right</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.mglvq67mg4o5">Delete selection</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.9vkm917kv6bk">Make sequence logo</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.pr3v49sxcqo3">Annotate selection</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.ksxtw4u2ma5j">Edit sequences</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.y7itip7i740o">Shade manually</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.h004upo4zakp">Block shading</a></span></p><p class="c3"><span class="c2 c9">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="#id.gbll90sv4vjh">Export selection</a></span></p><p class="c3"><span class="c11 c2 c9"><a class="c6" href="#id.gv6tcj9f2bsl">Troubleshooting and FAQ</a></span><hr style="page-break-before:always;display:none;"></p><a id="id.ctps4mkdca7y"></a><p class="c3"><span class="c16 c13 c2 c9">Introduction</span></p><a id="id.ltgos44kk4i8"></a><p class="c3"><span class="c2 c4">What is Slimshadey?</span></p><p class="c3"><span class="c2">SlimShadey is a light-weight Java-based tool that allows for the targeted analysis of sequences and offers direct residue level control in the editing, trimming, shading, and export of multiple sequence alignments. The extensive functionality implemented in SlimShadey takes advantage of JavaFX and includes complete editing (inserting, deleting, degapping, subsequence export), real-time shading (identity, similarity, dissimilarity, counts, matrix), annotation (domains, motifs, logos), and exporting of publication quality images in PNG and RTF format. &nbsp;This manual and </span><span class="c2 c15">associated videos</span><span class="c1">&nbsp;use the GLD-1 or CD44 protein sequences (reviewed in Lee and Schedl 2010 and Senbanjo and Chellaiah, 2017, respectively) to illustrate the major tools contained in the package. &nbsp;Additional sample sequences are provided to illustrate the range sequence alignments that can be manipulated. &nbsp;All of the calculation, shading, and editing of sample sequences is performed in real-time.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.lxv2blovq986"></a><p class="c3"><span class="c4 c2">Requirements</span></p><p class="c3"><span class="c2">SlimShadey will run without modification on Windows 7 (or higher), OS X / macOS, and Linux operating systems when Java 8 JRE or higher is installed (download from </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://www.oracle.com/java/technologies/javase-jre8-downloads.html&amp;sa=D&amp;source=editors&amp;ust=1619463442513000&amp;usg=AOvVaw24ISvMyLPtjCkwORJ9xbBK">Oracle</a></span><span class="c1">). &nbsp;Internet access is not required for SlimShadey, except for ScanProsite- and Hmmscan-based shading. &nbsp;SlimShadey will also run on most server platforms that support Java. &nbsp;For information on how to run SlimShadey on Linux or on a server please see the sections after the FAQ. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><hr style="page-break-before:always;display:none;"><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.1clwclydrbfq"></a><p class="c3"><span class="c4 c2">Quick start video tutorial</span></p><p class="c3"><span class="c1">To use this tutorial you will need Java 8 JRE installed, at least 2GB RAM, the &ldquo;SeqVerter.jar&rdquo; file downloaded, and 1MB free disk space. &nbsp;This video covers the following topics:</span></p><ul class="c10 lst-kix_reo1rr5lb8ia-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">Checking Java version</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Sample data provided shaded and unshaded</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Sequence input options for user data</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Sequence alignment using Opal</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Basic shading options</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Block shading and annotation</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Shading with a substitution matrix</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Shading with ScanProsite, Hmmscan, and regular expressions</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Sequence cropping</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Subsequences to new tab, degapping, and realignment</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Character editing</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Sequence logos</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Presentation / publication quality output</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Output to SlimShadey format</span></li></ul><p class="c3 c5"><span class="c1"></span></p><a id="id.f9y332j9rvnj"></a><p class="c3"><span class="c4 c2">Performance</span></p><p class="c3"><span class="c1">SlimShadey has minimal RAM requirements (2GB), does not require a dedicated graphics card, and will run on legacy systems provided they support Java 8 JRE. &nbsp;However, for larger datasets and a smoother experience, the availability of 16GB RAM, dedicated graphics card (for example, NVIDIA GTX 1070), and quad core processor or higher (for example, Intel Core i7-8700) will significantly improve performance. In addition, performance scales inversely with the number of characters displayed (i.e. the number of sequences x the number of characters per sequence); in larger alignments, selecting columns of interest and exporting them to a new tab, so that fewer characters are displayed, will dramatically improve performance. &nbsp;In addition, taking advantage of the fully reversible &ldquo;collapse identical sequences&rdquo; function (see below) can dramatically improve performance on legacy hardware when dealing with larger datasets with multiple identical sequences.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2">When it is not necessary to view the entire dataset on the screen or the rendering is not smooth during processing, additional performance can be gained by using lower the output resolution of your monitor. &nbsp;For example, using 720p (1280x720), 1080p (1920x1080), or 1440p (2650x1440) will result in smoother performance versus 4k (3840 x 2160) or higher. </span><span class="c1">&nbsp;For most applications, we recommend setting the screen output resolution to 1080p, however, any common resolution may be used as Slimshadey will scale the default window to &frac34; of the height and width of the users screen output resolution. </span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Our recommendations for large datasets (&gt;250 sequences x 500 characters) are as follows:</span></p><ul class="c10 lst-kix_kiw2208yjgf-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">CPU: Intel Core i7 8700K or AMD Ryzen 1700X</span></li><li class="c3 c8 li-bullet-0"><span class="c1">GPU: NVIDIA GTX 1070 or AMD Radeon RX 580</span></li><li class="c3 c8 li-bullet-0"><span class="c1">RAM: 16GB</span></li></ul><p class="c3"><span class="c1">For very large datasets, the memory allocation for the JVM may have to be altered (see FAQ). </span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Other than the text is not as clear as could be rendered on your system, there is no output penalty for working at lower screen resolutions. &nbsp;The resolution of the output alignment (as produced by the &lsquo;print preview&rsquo; option) is dictated by user-defined font size, and NOT the screen resolution. The output font size can be changed in the file menu in the print preview window if a high resolution printable or projectable image is desired. &nbsp;As such, SlimShadey&rsquo;s performance may be improved by reducing the screen resolution, without affecting the resolution of the final graphical output. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.yn88mcbs0yxp"></a><p class="c3"><span class="c16 c13 c2 c24">Checks to run before starting</span></p><p class="c3"><span class="c1">On your PC or server, from the terminal check your Java using &ldquo;java -version&rdquo;. &nbsp;This should return something similar to the the following:</span></p><p class="c3 c27"><span class="c7 c2">java version &quot;1.8.0_291&quot;</span></p><p class="c3 c27"><span class="c7 c2">Java(TM) SE Runtime Environment (build 1.8.0_291-b09)</span></p><p class="c3 c27"><span class="c7 c2">Java HotSpot(TM) 64-Bit Server VM (build 25.291-b09, mixed mode)</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">An example from using the Command Prompt from Windows 10 is below. &nbsp;Some of the version numbers (e.g. 291) will change based on updates. &nbsp;The same process is used to check the version through the Terminal on Mac OS X or Linux distributions.</span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 498.81px; height: 169.30px;"><img alt="" src="images/image1.png" style="width: 501.90px; height: 172.83px; margin-left: -0.00px; margin-top: -3.53px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c1">If your system meets or exceeds the requirements above then SlimShadey will operate on your system. &nbsp;If the &ldquo;java -version&rdquo; command does not work on your system but you are certain that you have installed Java and SlimShadey still runs then see the FAQ for an explanation. </span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.tzt37jgbtgkx"></a><p class="c3"><span class="c4 c2">Input file formats</span></p><p class="c3"><span class="c2">The preferred input file format for SlimShadey is sequential FASTA (aligned or unaligned). Alignments in other formats can easily be converted to FASTA using </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/&amp;sa=D&amp;source=editors&amp;ust=1619463442517000&amp;usg=AOvVaw0Wun1B98_pW8mG7SCG_pGc">SeqRet</a></span><span class="c2">&nbsp;(Madeira et al, 2019). For unaligned sequences, </span><span class="c1">SlimShadey has been packaged with OpalFX (Wheeler and Kececioglu, 2007) to allow for the rapid alignment of protein and nucleic acid sequences. SlimShadey supports the direct cut/paste of FASTA files from NCBI, Ensembl, and other databases. Multiple files can also be imported using drag and drop into SlimShadey&rsquo;s UI for convenience.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Example input format (partial GLD-1 FASTA OPAL aligned):</span></p><p class="c3 c5"><span class="c14 c2 c18 c31"></span></p><p class="c3"><span class="c7 c2">&gt;GLD-1_Caenorhabditis_latens</span></p><p class="c3"><span class="c7 c2">FGVTSQLETQNAESPSRSSILTPTSLEDETSPRKRFPLIETNISASDRWPPAPRRDGWSS</span></p><p class="c3 c5"><span class="c7 c2"></span></p><p class="c3"><span class="c7 c2">&gt;GLD-1_Caenorhabditis_briggsae</span></p><p class="c3"><span class="c7 c2">FGVSAQHENPSVDSPSRSSILTPTSLDDETSPRKSFQILESSVS-ADRWP-APRRDGWSS</span></p><p class="c3 c5"><span class="c7 c2"></span></p><p class="c3"><span class="c7 c2">&gt;GLD-1_Caenorhabditis_nigoni</span></p><p class="c3"><span class="c7 c2">FGVSAQHENPSVDSPSRSSILTPTSLDDETSPRKGFPILESSVS-TDRWP-APRRDGWSS</span></p><p class="c3 c5"><span class="c7 c2"></span></p><p class="c3"><span class="c2">SlimShadey also accepts the Shadey project file format (extension </span><span class="c14 c2">.mm</span><span class="c1">), which stores all sequence, annotation, consensus, and shading/styling information. &nbsp;This allows users to share their alignment markup/workflow and preferred session settings, to work on an alignment collaboratively or over multiple SlimShadey sessions.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.xge2wndr30p1"></a><p class="c3"><span class="c2 c15 c18">Supported formats </span><sup><a href="#cmnt1" id="cmnt_ref1">[a]</a></sup></p><p class="c3"><span class="c1">Other accepted input and output formats include Clustal, MSF, and Phylip 3.2. &nbsp;SlimShadey will extract the text, convert it to a FASTA file, and will discard any additional information. SlimShadey can also output to Clustal, MSF, and Phylip 3.2; however, only the sequence and sequence names will be saved (annotation and shading will be lost), and a consensus sequence, where appropriate. </span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2">Sequence names are appended to 10 characters when outputting to MSF and Phylip 3.2 to conform to the standards for each format. Similarly, Clustal sequence names are appended to 20 characters. &nbsp;The sequence names remain the same when outputting to FASTA.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.l8ltlbp36y9h"></a><p class="c3"><span class="c4 c2">How to input files to SlimShadey</span></p><p class="c3"><span class="c1">Files can be imported to SlimShadey in three ways. </span></p><ul class="c10 lst-kix_rddsjt44cy73-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">Aligned or unaligned sequences via a file explorer, accessible through the &lsquo;Start&rsquo; menu</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Aligned sequences can be dragged and dropped into SlimShadey&rsquo;s UI directly</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Individual FASTA-format alignments can be copied and pasted into SlimShadey via the &lsquo;Start&rsquo; menu</span></li></ul><p class="c3"><span class="c2">Note that multiple files can be input simultaneously via the file explorer and drag/drop; however, this does not apply when cut/paste of FASTA sequences is being used.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.eki70i1pyyw9"></a><p class="c3"><span class="c4 c2">SlimShadey Format (.mm)</span></p><p class="c3"><span class="c1">The .mm format has been designed to hold all of the information from a SlimShadey project. &nbsp; &nbsp;The text file holds all sequence information plus annotations and complete shading (including position, color, shade) information.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">File structure:</span></p><ul class="c10 lst-kix_u5mt6v244o5o-0 start"><li class="c3 c8 li-bullet-0"><span class="c2">The first 5 characters of a </span><span class="c14 c2">.mm</span><span class="c2">&nbsp;file are &ldquo;</span><span class="c14 c2">.slim</span><span class="c1">&rdquo;.</span></li><li class="c3 c8 li-bullet-0"><span class="c2">Data fields begin with &ldquo;</span><span class="c14 c2">$</span><span class="c2">&rdquo; followed by &ldquo;</span><span class="c14 c2">field_name</span><span class="c2">&rdquo; (e.g. </span><span class="c14 c2">fontsize</span><span class="c2">, </span><span class="c14 c2">collapse</span><span class="c2">, </span><span class="c14 c2">luminance</span><span class="c2">, etc.), and, within curly brackets (&ldquo;</span><span class="c14 c2">{</span><span class="c2">&ldquo; and &ldquo;</span><span class="c14 c2">}</span><span class="c2">&rdquo;) the data for that field. For example, to denote that the given alignment&rsquo;s UI font size is 24, a line in the file would be: </span><span class="c14 c2">$fontsize{24}</span></li></ul><ul class="c10 lst-kix_wulsapsxo6ef-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">Sequences and annotations are stored with one character per line, and separated by sequence names. For example, valid sequence data for the following FASTA alignment:</span></li></ul><p class="c3"><span class="c7 c2">&gt;seq1</span></p><p class="c3"><span class="c7 c2">AY</span></p><p class="c3"><span class="c7 c2">&gt;seq2</span></p><p class="c3"><span class="c14 c2">AW</span></p><ul class="c10 lst-kix_vknp6nujnm9e-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">With one empty (i.e. all space characters) annotation row, and shaded with white background and black text, the alignment would be saved as:</span></li></ul><p class="c3"><span class="c7 c2">$annotation_data{</span></p><p class="c3"><span class="c7 c2">%EMPTY_ANNO_1</span></p><p class="c3"><span class="c7 c2">&nbsp;.#FFFFFF.#000000</span></p><p class="c3"><span class="c2 c7">&nbsp;.#FFFFFF.#000000</span></p><p class="c3"><span class="c7 c2">}</span></p><p class="c3"><span class="c7 c2">$sequence_data{</span></p><p class="c3"><span class="c7 c2">%seq1</span></p><p class="c3"><span class="c7 c2">A.#FFFFFF.#000000</span></p><p class="c3"><span class="c7 c2">Y.#FFFFFF.#000000</span></p><p class="c3"><span class="c7 c2">%seq2</span></p><p class="c3"><span class="c7 c2">A.#FFFFFF.#000000</span></p><p class="c3"><span class="c7 c2">W.#FFFFFF.#000000</span></p><p class="c3"><span class="c7 c2">}</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.sd310g36q790"></a><p class="c3"><span class="c4 c2">Sample data provided</span></p><p class="c3"><span class="c1">Multiple aligned and unaligned sample files are provided to illustrate functionality. &nbsp;Aligned sequences are ported directly to SlimShadey, whereas unaligned sequences are passed directly to Opal for alignment.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">The videos below walk through the basics of how to use SlimShadey.</span></p><p class="c3"><span class="c1 c15">Link to video</span></p><p class="c3 c5"><span class="c1 c15"></span></p><a id="id.6tj099fx7d07"></a><p class="c3"><span class="c4 c2">Parameter file</span></p><p class="c3"><span class="c1">SlimShadey always initializes with default shading rules for each method (i.e. ClustalX, ScanProsite, RASMOL, etc.), even if a .slim file is edited by a different user who has changed their default settings. Note that the previous user&rsquo;s shading can be retained during input. For example, when clicking RASMOL shading, choose &lsquo;Retain&rsquo; when prompted to clear shading, and their shading will still be accessible; however, your RASMOL remote will open with your default settings. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">For users that require custom default shading, the package includes a parameter file that can be easily modified with a text editor to change default rules for each alphabet without the need for recompiling. &nbsp;For example, new amino acid groups can be defined, the matrices can be modified or new ones added, consensus rules changed, default coloring modified, and non-standard characters can be added. &nbsp;While default coloring RGB values (in hex format) and consensus rules can be changed without any issues, if you add a new character, it must be represented in each substitution matrix. &nbsp;</span></p><p class="c3 c5"><span class="c1 c15"></span></p><p class="c3"><span class="c1 c15">For a more detailed description see the following video.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Unzip the .jar file, and go to the sub-directory:</span></p><p class="c3"><span class="c7 c2">[edu/tcnj/seqverter/sequence/]</span></p><p class="c3"><span class="c2">In the </span><span class="c14 c2">[.../sequence/] </span><span class="c2">directory, there are </span><span class="c14 c2">*.salpha</span><span class="c2">&nbsp;files, which contain the rules. For example, to change the rules for protein sequences, the user can edit the </span><span class="c14 c2">Protein.salpha</span><span class="c1">&nbsp;file.</span></p><p class="c3 c5"><span class="c16 c13 c2 c9"></span></p><a id="id.hm2xctozcu7z"></a><p class="c3"><span class="c16 c13 c2 c9">File menu options</span></p><a id="id.517oqinszmgw"></a><p class="c3"><span class="c4 c2">Rename tab</span></p><p class="c3"><span class="c1">Allows the user to rename the selected tab. Custom tab names allow for streamlining of workflow, by specifying the contents of each open alignment, and giving a default filename when the project is exported to FASTA, RTF, PNG, or the Shadey project format. &nbsp;Each tab is treated as a different project and must be saved separately. &nbsp;The number of tabs that can be open during a session is limited by total memory allocated to the JVM (see troubleshooting). &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.alvel8j40h1v"></a><p class="c3"><span class="c4 c2">Realign sequences (align sequences)</span></p><p class="c3"><span class="c1">SlimShadey is provided with OpalFX, an interface to the Opal multiple sequence alignment algorithm (Wheeler, 2007), as a fast, accurate, and convenient sequence alignment tool. &nbsp;OpalFX can be used to align unaligned sequences and to realign a multiple sequence alignment already loaded into a tab. OpalFX also supports translation and alignment of nucleotide sequences with all 30+ known genetic codes, as well as codon-delimited alignment. &nbsp;Once alignments have been completed, the user can directly import the results to SlimShadey or save the alignment in FASTA format from the file menu.</span></p><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.e6t9n9f2hum4"></a><p class="c3"><span class="c4 c2">Print preview</span></p><p class="c3"><span class="c1">Print preview allows for WYSIWYG formatting and rendering of publication quality images. The multiple sequence alignment is organized into rows, and the user is given control over numbering, truncation of names, font size, number of columns per row, and whether to include annotations or a consensus sequence. The output can be saved as a PNG file for downstream analysis for further annotation in image processing software. &nbsp;</span></p><p class="c3 c5"><span class="c19 c2 c18"></span></p><ul class="c10 lst-kix_90sjlybkt4o7-0 start"><li class="c3 c8 li-bullet-0"><span class="c2 c18">NOTE: </span><span class="c1">PNG resolution is based on font size. Increasing the font size will increase the output resolution. As such, if additional resolution is necessary then font size can be adjusted to meet your requirements.</span></li></ul><p class="c3 c5"><span class="c1"></span></p><a id="id.dvl2g4twj8lk"></a><p class="c3"><span class="c4 c2">Output to RTF (rich text format)</span></p><p class="c3"><span class="c1">Exporting the contents of a tab in RTF format allows the output of SlimShadey to be opened and edited manually in word processing software (e.g. MS Word, OpenOffice, Google Docs). Exporting to RTF allows users to directly paste a styled/formatted alignment into other documents. This format also allows the user to change (add, insert, and delete) characters after the contents of a tab have been output without using the SlimShadey interface. &nbsp;As output options, the user is provided control over font size, name truncation (i.e. number of characters in name), and whether to include annotations or numbering.</span></p><p class="c3 c5"><span class="c1"></span></p><ul class="c10 lst-kix_d0pnlbhmephp-0 start"><li class="c3 c8 li-bullet-0"><span class="c2 c18">NOTE: </span><span class="c1">The RTF is saved with all color and formatting specified exactly as in the SlimShadey tab. Some word processors may not display the RTF properly. For best results we recommend Microsoft Word.</span></li></ul><p class="c3 c5"><span class="c1"></span></p><a id="id.d6fymjflnuw"></a><p class="c3"><span class="c4 c2">Save alignment as text</span></p><p class="c3"><span class="c2">SlimShadey allows any tab to be exported to a flat text file for use in other applications. &nbsp;The pulldown menu under the file name: &ldquo;Save as type&rdquo; allows selection of multiple supported formats for export. &nbsp;Additional formats can subsequently be generated using </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/&amp;sa=D&amp;source=editors&amp;ust=1619463442526000&amp;usg=AOvVaw0qdKWScehnXkT0fpXCPDZv">SeqRet</a></span><span class="c2">&nbsp;(Madeira et al, 2019).</span></p><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.hpjwed9l5clo"></a><p class="c3"><span class="c2 c9">Save as project</span></p><p class="c3"><span class="c2">SlimShadey outputs projects to the .mm format, allowing complete sessions to be shared. The &ldquo;</span><span class="c2 c18">.mm&rdquo; </span><span class="c1">format saves the alignment along with all shading, annotations and consensus sequence, session settings, including UI font size, the first index of the alignment, luminance threshold for text coloring, the alphabet the alignment (RNA, DNA, or protein), and whether to collapse identical sequences.</span></p><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.fj721mwizur4"></a><p class="c3"><span class="c2 c9 c13">Edit menu options</span></p><a id="id.bqbga6li8nbt"></a><p class="c3"><span class="c4 c2">Session settings</span></p><a id="id.grx1d1qdwxnj"></a><ul class="c10 lst-kix_hefi1acv3kbb-0 start"><li class="c3 c8 li-bullet-0"><span class="c2 c18">First index:</span><span class="c1">&nbsp;Sets the index of the first column of the alignment to the specified value. By default, the first column is &lsquo;1&rsquo;. &nbsp;For example, if you have a portion of a larger sequence, you may want to change this value to better reflect the position of the sequence.</span></li></ul><a id="id.9yvme81q8cbt"></a><ul class="c10 lst-kix_hefi1acv3kbb-0"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Font size:</span><span class="c1">&nbsp;Changes the UI font size to the specified value. Font size is 18px by default, to display as much of the alignment as clearly as possible on most machines. &nbsp;SlimShadey renders all characters in real-time so changes in font size will change the number of characters displayed on the screen and may alter the performance of the UI.</span></li></ul><a id="id.av7fmqwu0if0"></a><ul class="c10 lst-kix_hefi1acv3kbb-0"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Luminance threshold: </span><span class="c1">&nbsp;The luminance of a color can be calculated from its RGB color space. Depending on whether the shading color&rsquo;s luminance is above or below the threshold value, the text color is set to either black or white. The user can change the text color to any RGB value manually, but built-in shading/coloring algorithms will only color text black or white when shading is performed, depending on the background shading assigned. After shading is performed, the user has residue-level control over the color of the letter and the background. &nbsp;The luminance threshold is by default 0.179, which generally assigns white or black text color appropriately based on the background shading color. Changing this value is not recommended.</span></li></ul><a id="id.xnfioacy2ouj"></a><ul class="c10 lst-kix_hefi1acv3kbb-0"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Hover refresh rate:</span><span class="c2">&nbsp;This number (in milliseconds) ranges from 0 to 999, and is 250ms by default. &nbsp;Lower refresh rates require more robust hardware as calculations will be performed more frequently. &nbsp;We recommend 250ms or greater for systems using integrated graphics. If you notice substantial lag in hovering over residues in the alignment, increasing this value may help reduce this lag, lower system load, and improve performance. It allows the user to fine tune the interface to match their system capabilities.</span></li></ul><p class="c3 c5"><span class="c16 c13 c2 c9"></span></p><a id="id.tpunrwq89nsk"></a><p class="c3"><span class="c4 c2">Gap only columns</span></p><p class="c3"><span class="c1">SlimShadey can identify gap only columns so they can be removed by the user or avoided prior to exporting subsequences.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.a3ggyvgw0br"></a><p class="c3"><span class="c4 c2">Calculate consensus</span></p><p class="c3"><span class="c2">SlimShadey can calculate three types of consensus sequence (Simple, Dots and Identify, and ClustalX). &nbsp;Simple consensus outputs a lowercase character for majority in a column and an uppercase when characters are identical in a column. Dots and identify uses the ClustalW rule set to calculate the consensus (Thompson, 1994). ClustalX consensus uses the rule set from Jeanmougin et al., 1998. These can be viewed in </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=http://www.clustal.org/download/clustalx_help.html&amp;sa=D&amp;source=editors&amp;ust=1619463442528000&amp;usg=AOvVaw15lDhC0RCCM9KZStJlEGGw">ClustalX&rsquo;s manual</a></span><span class="c1">, under the &lsquo;Color&rsquo; heading.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.9ve34sqice01"></a><p class="c3"><span class="c4 c2">Add annotation row</span></p><p class="c3"><span class="c2">This function allows the user to add an annotation row. For detailed instructions on editing, moving, deleting, or adding annotations, please refer to the section on </span><span class="c2 c9">Directly Editing the Alignment</span><span class="c2">. Annotation rows are useful for highlighting features of the alignment, such as regions with conserved structure, outstanding residues, insertions, or deletions of interest. Annotations can be added by individual column or by region (block annotation over multiple columns). &nbsp;For an annotation over a single column, right-click above the region of interest and select &ldquo;Annotate selection&rdquo; from the pull-down menu. &nbsp;If you want to annotate a region, right-click-and-drag over the region desired and select &ldquo;Annotate selection&rdquo;. &nbsp;Commonly used characters are available for copy/paste use in the annotation window. &nbsp;For regional annotation the single character will be copied over all columns. &nbsp;For example, to define an alpha helical region, the user can cut-paste the &lsquo;</span><span class="c14 c2">&alpha;</span><span class="c2">&rsquo;, which will be copied over all columns selected. &nbsp;If multiple characters are used then the characters will repeat over the region, e.g if 4 columns are selected, and &lsquo;</span><span class="c14 c2">&alpha;</span><span class="c2">&rsquo; is entered, the annotation &lsquo;</span><span class="c14 c2">&alpha;&alpha;&alpha;&alpha;</span><span class="c2">&rsquo; will be applied. &nbsp;</span></p><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.vlmi6y94qjfg"></a><p class="c3"><span class="c4 c2">Collapse identical sequences</span></p><p class="c3"><span class="c1">This function collapses identical sequences into groups, which can be unpacked when analysis has been completed without any loss of data. This function works by performing an all-against-all comparison for identify and only displays unique sequences. When a group of identical sequences is collapsed into a group, only the shading and text color of the first sequence is saved; when uncollapsing the sequences, the shading that was applied to the collapsed sequence is applied to all constituent sequences.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">SlimShadey performs real-time calculations on the alignment as rendered. &nbsp;For example, if there are multiple identical sequences that result in the over representation of characters, when the collapse function is called and they are reduced to a single sequence, the same characters may now be underrepresented. &nbsp;More specifically, if you have 5 sequences, 3 of which are identical, then the 3 identical sequences will be a majority when uncollapsed, and will not be a majority after collapsing.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">When working with large datasets that may have many identical sequences, using this function can improve the performance of SlimShadey. For example, rapidly identifying low frequency polymorphisms in a population benefits from not rendering uninformative sequences. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.9zckf9ky6mjt"></a><p class="c3"><span class="c13 c2 c9">Shade menu options</span></p><a id="id.52y827syhtee"></a><p class="c3"><span class="c2 c9">Clear shading</span></p><p class="c3"><span class="c1">This function clears all shading and returns the input sequence to the default settings.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.b9onwk15hiku"></a><p class="c3"><span class="c4 c2">Shade by unique residues</span></p><p class="c3"><span class="c2">To reveal outstanding residues, this function highlights unique residues in a column. &nbsp;For flexibility, the remote allows adjusting the maximum number of residues in the column. &nbsp;This allows restricting shading to those residues within a column whose count is below a threshold count. &nbsp;This is an easy and sensitive method for determining interesting polymorphisms in sequence alignments. The ability to store logs of differing residues by position, and visualize the locations and apparent densities of these differences, may be helpful in directing further analysis and generating testable hypotheses.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.9gn34rpv1t23"></a><p class="c3"><span class="c4 c2">Shade by residue frequency</span></p><p class="c3"><span class="c1">This function allows shading to a simple majority within columns. The threshold frequency for calculating the consensus can be adjusted using the slider, to between 0.5 and 1, to change shading in real-time. &nbsp;Within a column, if the proportion of a single residue exceeds this threshold frequency, the residue is shaded black to denote identity, and similar residues gray to denote similarity; if the total proportion of different, but physically or chemically similar residues exceeds the threshold, those similar residues are shaded gray to denote similarity (e.g. if in a column neither leucine nor isoleucine alone exceed the threshold frequency, but together do, both are shaded &lsquo;gray&rsquo;). A threshold between 0.5 and 1 can be set by the user. The lower limit of 0.5 is set to avoid determining more than one consensus residue within a column. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2">This color scheme can be inverted (Inverse shading) to produce &lsquo;inverse shading,&rsquo; in which residues to be shaded black are left white, residues to be left white are shaded black, and residues to be shaded grey are shaded grey; color inversion is particularly useful for demonstrating dissimilarity or lack of conservation.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.fmgkz650ryq6"></a><p class="c3"><span class="c4 c2">Shade by sequence</span></p><p class="c3"><span class="c1">Shade by sequence allows for shading matches, similarities, and differences relative to a master sequence selected by the user. &nbsp;The master sequence can be any of the sequences in the alignment. &nbsp;By default, the master sequence is not shaded; however, this can be changed to any color from the remote. &nbsp;This function is useful for revealing identity to a specific lineage.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.gvklkfo29x3u"></a><p class="c3"><span class="c4 c2">Shade with substitution matrix</span></p><p class="c3"><span class="c1">Sequences can be shaded with a relative match to one of the BLOSUM matrix series (30-100), Gonnet250, PAM250, or identity. &nbsp;For each substitution matrix, a shading matrix is calculated based on two RGB values, one for a perfect match and one for the worst possible match. For each residue, this corresponds to the maximum (i.e. identity) and minimum score possible, respectively. For example, when using BLOSUM62, and comparing a residue to alanine (A), the maximum score is when aligned to another A, at +4. &nbsp;This represents a perfect match and is assigned the match color. &nbsp;Similarly, using BLOSUM62, alanine aligned with tryptophan (W) results in the minimum score of -3, and produces shading closest to the mismatch color. &nbsp;As the residue changes from the worst possible match, and the score of the aligned residue increases, the RGB values scales linearly to the match color. Comparisons to a gap character (-) always result in no shading.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.t7fqmdtg27dm"></a><p class="c3"><span class="c4 c2">Custom Prosite regex shading</span></p><p class="c3"><span class="c2">SlimShadey allows shading of sequences using matches to regular expressions using SwissProt&rsquo;s ScanProsite syntax. &nbsp;The SwissProt regular expression syntax can be downloaded </span><span class="c11 c2"><a class="c6" href="ftp://ftp.expasy.org/databases/prosite/profile.txt">here</a></span><span class="c2">&nbsp;or viewed </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://prosite.expasy.org/scanprosite/scanprosite_doc.html%23mo_motifs&amp;sa=D&amp;source=editors&amp;ust=1619463442532000&amp;usg=AOvVaw1xPzIAqT_01lV0TGVQgc8k">here</a></span><span class="c2">. Each sequence in the alignment is scanned for a match to the pattern, regardless of position, and matches are previewed in a graphical window. &nbsp;Any matches can be applied to the alignment and viewed directly on the sequence alignment. &nbsp;SlimShadey&rsquo;s scanner disregards gap characters, e.g. the pattern &lsquo;</span><span class="c14 c2 c22">MP</span><span class="c2">&rsquo; will match a subsequence given by &lsquo;</span><span class="c14 c2">...AR</span><span class="c14 c2 c22">M</span><span class="c14 c2">--</span><span class="c14 c2 c22">P</span><span class="c14 c2">LC...</span><span class="c2">&rsquo; by ignoring gaps. &nbsp;In addition, the scanner supports the use of multiple regular expressions simultaneously, which allows for the mapping of multiple motifs in a single scan. &nbsp;Using the extended ScanProsite syntax, simple strings can be rapidly searched. For example, &lsquo;</span><span class="c14 c2">MPSC</span><span class="c2">&rsquo; is the same as the full expression in ScanProsite syntax, &lsquo;</span><span class="c14 c2">M-P-S-C</span><span class="c1">&rsquo;.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.4i3na39zeuwm"></a><p class="c3"><span class="c4 c2">Prosite-based database shading</span></p><p class="c3"><span class="c1">Prosite-based shading retrieves matches from prosite and renders them on the alignment (Hulo, 2004; Sigrist, 2013). &nbsp;The default implementation uses ExPASy&rsquo;s standard settings and filters low complexity patterns and includes ProSite profiles in the output. &nbsp;The hits can be applied to the alignment, where all related domains will be rendered in the same color. &nbsp;Information about the pattern, including ScanProsite hit documentation, can be retrieved by clicking on the colored region of the diagram. &nbsp;Sequences are degapped prior to scanning so that hits will be rendered even if there are gaps or the hits are present at different positions.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.g8bqwfhft4s6"></a><p class="c3"><span class="c4 c2">hmmscan based shading</span></p><p class="c3"><span class="c2">The hmmer sequence analysis package implements an algorithm for finding motifs and domains within sequences using profile HMMs (Potter et al., 2018). SlimShadey uses an online implementation of hmmscan, hosted at the EBI (</span><span class="c2 c11"><a class="c6" href="https://www.google.com/url?q=https://www.ebi.ac.uk/Tools/hmmer/search/hmmscan&amp;sa=D&amp;source=editors&amp;ust=1619463442533000&amp;usg=AOvVaw3tgK_eJH1RUxc6vGW0eS8h">https://www.ebi.ac.uk/Tools/hmmer/search/hmmscan</a></span><span class="c2">) to identify motifs in each sequence in the alignment. Motifs are all from the Pfam database (Potter et al., 2018), and documentation can be retrieved for these within SlimShadey. By default, all hits to the same accession are assigned the same RGB value for shading. &nbsp;Each motif&rsquo;s default color can be changed independently, and individual motifs can be shown or hidden on the alignment. &nbsp;All positional information is carried from hmmscan to SlimShadey and retained on the alignment, even when gaps are present within a motif.</span></p><p class="c3 c5"><span class="c1"></span></p><a id="id.qa7xocpx59fi"></a><p class="c3"><span class="c4 c2">Default RASMOL coloring</span></p><p class="c3"><span class="c1">Default coloring of alignments uses the RASMOL color scheme (Sayle et al, 1995), which uses contrasting colors to distinguish residues with different physico-chemical properties.</span></p><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.6vr4ddq4bwzs"></a><p class="c3"><span class="c4 c2">Default ClustalX coloring</span></p><p class="c3"><span class="c2">This function calculates a consensus and applies shading using the rules from ClustalX for protein sequences (Larkin, 2007). &nbsp;Each amino acid is shaded according to its identity, and the consensus character for its column. The rules for this are described in the &lsquo;</span><span class="c14 c2">colorprot.par</span><span class="c1">&rsquo; parameter file distributed with ClustalX.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2 c9">NOTE:</span><span class="c1">&nbsp;If you plan to shade in multiple layers, the dynamic coloring and shading of characters implemented in JavaFX requires a specific order of operations.</span></p><ol class="c10 lst-kix_1pwbqchre05h-0 start" start="1"><li class="c3 c8 li-bullet-0"><span class="c1">Use either: shade by unique residues, shade by residue frequency, shade by sequence, default RASMOL coloring, or default ClustalX coloring. Updating the properties of these shading modalities will overwrite all shading in the alignment.</span></li><li class="c3 c8 li-bullet-0"><span class="c2">Once one of the aforementioned shading modes have been used as a &lsquo;base&rsquo; for the alignment, domains or motifs may be highlighted by using: shade with Prosite Motifs.</span></li></ol><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.vdf66cgu6ys3"></a><p class="c3"><span class="c13 c2 c9 c16">Directly Editing the Alignment</span></p><p class="c3"><span class="c2">In SlimShadey, an alignment is split into two panes. The left pane holds names of the sequences and names of the annotation rows. &nbsp;The right pane holds the alignment and annotations. Clicking on a sequence or annotation row&rsquo;s name, and left- or right-clicking somewhere on the alignment, will open a menu with options to manipulate properties of the alignment.</span></p><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.8ya7dcerna4n"></a><p class="c3"><span class="c4 c2">Sequence-level control (left pane, clicking on sequence name or annotation name)</span></p><p class="c3"><span class="c2">In an alignment in SlimShadey, the left pane</span><span class="c2">&nbsp;has the names of the annotations and sequences. Clicking on the name of a sequence or annotation allows the user to rename, reorder, or delete the selection.</span></p><a id="id.48n4ln4lu5z3"></a><ul class="c10 lst-kix_cbfpvidehrpt-0 start"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Annotations: </span><span class="c1">Clicking on an annotation allows the user to edit the annotation row&rsquo;s name, add a new annotation row, change the relative position of an annotation row (up or down), and delete an annotation row.</span></li></ul><a id="id.jor39ylau6a2"></a><ul class="c10 lst-kix_cbfpvidehrpt-0"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Sequence names: </span><span class="c1">Similar to annotation rows, sequences can be renamed, reordered, or deleted by clicking on the name. &nbsp;Sequences can also be reordered relative to a chosen sequence, based on pairwise scoring between every other sequence and the chosen sequence with a substitution matrix - the chosen sequence is moved to the top, and sequences are reordered below in order of highest- to lowest-scoring.</span></li></ul><a id="id.aoyjfs1u5b2e"></a><ul class="c10 lst-kix_cbfpvidehrpt-0"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Consensus: </span><span class="c2">If changes are made to the sequence alignment without realignment, the consensus can be recalculated by clicking on &ldquo;</span><span class="c14 c2">Consensus</span><span class="c2">&rdquo; in the left pane at the bottom of the sequence alignment. &nbsp;If the sequences are realigned, the consensus sequence is automatically recalculated based on simple majority.</span></li></ul><p class="c3 c5"><span class="c4 c2"></span></p><a id="id.q0a5d6dxwhgo"></a><p class="c3"><span class="c4 c2">Active functions (right pane, left-click or right-click on alignment)</span></p><p class="c3"><span class="c1">SlimShadey allows for residue level control of editing, shading, and recoloring of sequence alignments. &nbsp;Custom editing and shading can also be used with annotations. &nbsp;These options can be accessed by right- or left-clicking directly on a character in the alignment. &nbsp;This includes the following functions:</span></p><a id="id.dm56sspqkdti"></a><ul class="c10 lst-kix_yoy1ormm4wf-0 start"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Residue-level control:</span><span class="c1">&nbsp;editing (changing, deleting, inserting) &nbsp;individual residues and their properties (color, shade)</span></li></ul><a id="id.ajyykno5oph5"></a><ul class="c10 lst-kix_yoy1ormm4wf-0"><li class="c3 c8 li-bullet-0"><span class="c2 c18">Column-level control: </span><span class="c1">deleting individual columns or groups of columns, editing properties of groups of columns, moving subsequences to new tabs for downstream analysis, hading subsequences to indicate new motifs, domains, or regions</span></li></ul><p class="c3"><span class="c1">The active functions are implemented as follows:</span></p><a id="id.j0ntpg7wy8e7"></a><ul class="c10 lst-kix_weixxh5ko91p-0 start"><li class="c3 c8 li-bullet-0"><span class="c2 c9">Left-click</span><span class="c1">&nbsp;on sequence alignment activates the residue level edit function. &nbsp;This allows for the following processes: &nbsp;</span></li></ul><a id="id.4hsb458l6kzn"></a><ul class="c10 lst-kix_weixxh5ko91p-1 start"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Edit: </span><span class="c1">From the edit strip at the bottom of the window the properties of the character can be changed. &nbsp;These properties include the character identify, shading color, and text color can be modified. </span></li></ul><a id="id.h3u7kqqh0hxy"></a><ul class="c10 lst-kix_weixxh5ko91p-0"><li class="c3 c8 li-bullet-0"><span class="c2 c9">Right-click</span><span class="c1">&nbsp;on the sequence alignment activates the column editing function. &nbsp;This allows for the following processes:</span></li></ul><a id="id.y1pi0a97bnyw"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Insert left/right</span><span class="c1">: Allows for the insertion of a new column to the left or right of the position identified by the right-click (green shade column indicator).</span></li></ul><a id="id.mglvq67mg4o5"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Delete selection:</span><span class="c2">&nbsp; This allows the d</span><span class="c1">eletion of individual columns or groups of columns defined with the right-click-and-drag functionality.</span></li></ul><a id="id.9vkm917kv6bk"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Make sequence logo: </span><span class="c2">Sequences that are defined by </span><span class="c1">Generation of sequence logos from subsequences right-click-and-drag can be converted to sequence logo &nbsp;in PNG format that can be printed or inserted in presentations.</span></li></ul><a id="id.pr3v49sxcqo3"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Annotate selection: </span><span class="c2">Allows block annotation of selected subsequences (see above, Add annotation row</span><span class="c4 c2">)</span></li></ul><ul class="c10 lst-kix_weixxh5ko91p-2 start"><li class="c3 c23 li-bullet-0"><span class="c2">If 5 columns are selected and the user enters the annotation &lsquo;</span><span class="c14 c2">A</span><span class="c2">&rsquo;, then the annotation produced will extend the pattern to &lsquo;</span><span class="c14 c2">AAAAA</span><span class="c2">&rsquo;. Similarly, if the user enters &lsquo;</span><span class="c14 c2">AB</span><span class="c2">&rsquo;, the annotation will be &lsquo;</span><span class="c14 c2">ABABA</span><span class="c2">&rsquo;. This allows quickly adding an annotation for multiple columns with a fixed sequence, e.g. &lsquo;</span><span class="c14 c2">&rArr;&rArr;...&rArr;&rArr;</span><span class="c2">&rsquo; by selecting those columns, </span></li></ul><a id="id.ksxtw4u2ma5j"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Edit sequences</span><span class="c2">: This function allows for the c</span><span class="c1">olumn level editing of individual residues</span></li></ul><ul class="c10 lst-kix_weixxh5ko91p-2 start"><li class="c3 c23 li-bullet-0"><span class="c2">Like the option above, a smaller subsequence can be entered and extended (e.g. the user can select 3 columns, enter &lsquo;</span><span class="c14 c2">A</span><span class="c2">&rsquo; for a sequence, and that sequence&rsquo;s characters will be &lsquo;</span><span class="c14 c2">AAA</span><span class="c1">&rsquo; in those columns). Alternatively, the user can choose to infer </span></li></ul><a id="id.y7itip7i740o"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Shade manually: </span><span class="c2">Manual shading provides c</span><span class="c1">ontrol of shading individual sequences</span></li></ul><a id="id.h004upo4zakp"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Block shading: </span><span class="c1">Fully manual control of shading over motifs, domains, novel regions, and sections of annotations</span></li></ul><a id="id.gbll90sv4vjh"></a><ul class="c10 lst-kix_weixxh5ko91p-1"><li class="c3 c12 li-bullet-0"><span class="c2 c18">Export selection: </span><span class="c1">The user can select right-click-and-drag to identify the subsequences to export. &nbsp;The selected sequence, with the names retained, will be exported to a new tab that is independent from the original. &nbsp;New shading rules and additional daughter tabs can be generated as desired and saved independently.</span></li></ul><p class="c3"><span class="c2 c9">NOTE: </span><span class="c1">An editing warning is included to warn the user of unintended changes that may alter the information content in the alignment. &nbsp;The user can turn off the warning and free-edit the sequence alignment to make multiple changes in sequence. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3 c5"><span class="c1"></span></p><hr style="page-break-before:always;display:none;"><p class="c3 c5"><span class="c16 c13 c2 c9"></span></p><a id="id.gv6tcj9f2bsl"></a><p class="c3"><span class="c16 c13 c2 c9">Troubleshooting and FAQ</span></p><p class="c3"><span class="c19 c2 c18">What if SlimShadey fails to launch?</span></p><p class="c3"><span class="c1">Use the terminal and use the &ldquo;java -version&rdquo; command to check your version of Java JRE. &nbsp;SlimShadey requires Java 1.8.0 or higher.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">What if you have checked your Java version and SlimShady still does not launch?</span></p><p class="c3"><span class="c2">In rare cases, Java JRE may need to be uninstalled and reinstalled. &nbsp;Follow the instructions for uninstalling all Java components </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://www.java.com/en/download/help/uninstall_java.xml&amp;sa=D&amp;source=editors&amp;ust=1619463442538000&amp;usg=AOvVaw1F_Cx-UuL3C6FyRAQJc0Cl">here</a></span><span class="c2">, reboot your system, reinstall Java (</span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://www.oracle.com/java/technologies/javase-jre8-downloads.html&amp;sa=D&amp;source=editors&amp;ust=1619463442539000&amp;usg=AOvVaw1HKQt7Pw8QqBEGOuT4dvaO">Oracle</a></span><span class="c2">), and reboot your system again</span><span class="c1">.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">The &ldquo;java -version&rdquo; command fails but SlimShadey still runs. &nbsp;Why is that?</span></p><p class="c3"><span class="c2">The java -version command simply allows you to get information about the version of Java you have installed. &nbsp;When the command fails to return the expected version information it simply means the path has been corrupted in some way. &nbsp;SlimShadey will run even if this command fails if you have a recent version of Java installed. &nbsp;As such, the java -version command failing is not an issue if SlimShadey runs; however, tf you would like to fix this error then download the </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://java.com/en/download/uninstalltool.jsp&amp;sa=D&amp;source=editors&amp;ust=1619463442539000&amp;usg=AOvVaw0qh08Jpjhc1CsiM8BSioKq">Java Uninstall Tool</a></span><span class="c2">&nbsp;from Oracle and uninstall all versions of Java that you have currently installed and reboot. &nbsp;Once you have fully uninstalled Java and rebooted, reinstall Java to restore the path variable. &nbsp;You can also fix this error by directly installing or editing the path variable.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">What if SlimShadey is running slowly?</span></p><p class="c3"><span class="c1">Slimshadey has been designed to run on legacy systems; however, it will run more slowly on systems that only support 32-bit versions of Java. &nbsp;Significant performance increases can be obtained by moving to hardware that can support 64-bit versions. &nbsp;The performance of SlimShadey can also be improved by using a system with at least 8GB of memory and a dedicated GPU. &nbsp;If larger datasets are being processed then SlimShadey will require additional memory (&gt;16GB) and faster CPU (e.g. Intel Core i7 or higher).</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">What if SlimShadey is running slowly on a dual monitor system?</span></p><p class="c3"><span class="c2">This occurs when one monitor is using the integrated graphics (e.g. Intel i5 with integrated HD graphics) and a second monitor is driven by a dedicated graphics card. &nbsp;In this case, SlimShadey may run significantly lower if it is rendering on the monitor supported by the integrated graphics. &nbsp;The solution is to run both monitors off the dedicated graphics card if possible or move the SlimShadey window to the monitor supported by the dedicated graphics card. &nbsp;For Windows 10 laptops with multiple displays, you can configure these properties in Advanced display settings and specify the display for each monitor. </span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2 c18">What if SlimShadey is running slowly on a system with updated hardware?</span></p><p class="c3"><span class="c1">Depending on your system configuration you may have to increase the memory available to SlimShadey for better performance. &nbsp;Using the terminal from the SlimShadey directory launch the software using the following command:</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c7 c2">java -Xmx16g -Xms16g -jar SlimShadey.jar</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">16g is the memory allocated to the JVM gigabytes. &nbsp;This value can be increased to the limit of the memory available on your machine (e.g. 32g, 64g, etc.). &nbsp;If you have less than 8GB of memory then use the following:</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c7 c2">java -Xmx2048m -Xms2048m -jar SlimShadey.jar</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">2048m is the memory allocated to the JVM in megabytes (or 2 gigabytes). &nbsp;This value can be increased to the limit of the memory available on your machine in multiples of 1024.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">How many tabs can I have open?</span></p><p class="c3"><span class="c1">We have not restricted the number of tabs that can be open at one time. &nbsp;Each tab is treated as a separate project but only the active tab is rendered in real-time. &nbsp;Depending on your system resources and memory allocation, SlimShadey may run more slowly if a large number of tabs are open. &nbsp;Since only the active tab is rendered in real time, having multiple tabs open does not impact the performance of SlimShadey unless system resources are limited.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Can I open multiple instances of SlimShadey?</span></p><p class="c3"><span class="c1">Yes. &nbsp;It is possible to open multiple instances of SlimShadey, however, the additional rendering may compromise the performance of real-time functions. &nbsp;If you are working with multiple sequences, we recommend using different tabs in a single instance for the best performance.</span></p><p class="c3 c5"><span class="c19 c2 c18"></span></p><p class="c3"><span class="c2 c18 c19">What if I have a large dataset?</span></p><p class="c3"><span class="c1">If you are working with a large dataset then you may need to increase the memory allocated to the JVM (Java Virtual Machine). &nbsp;See troubleshooting on how to get better performance on updated hardware (above). &nbsp;If you have access to a server with sufficient allocation of resources then you may be able to run SlimShadey without any input lag with extremely large datasets.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2 c18">Is there a limit to the number of sequences SlimShadey can handle?</span></p><p class="c3"><span class="c1">Slimshadey was designed for the laboratory scale analysis of sequence data; however, we have not limited the number of sequences that can be processed. &nbsp;The upper limit of sequence that can be analyzed will be determined by your system hardware, memory available to the JVM, and the type of processing (visualization vs alignment). If you have a genome scale dataset then you will have to use a different software package. &nbsp;Based on the sample system used for this work (Intel Core i7 8750H, 16GB RAM, NVIDIA GTX 1060), 200 sequences with approximately 1000 characters each can be processed without significant lagging. &nbsp;Larger datasets or less robust hardware may result in some lag in the real-time processing functions.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2 c18">Do I have to use OpalFX to align my sequences?</span></p><p class="c3"><span class="c1">No. SlimShadey will accept any aligned or unaligned text file in FASTA format. We have provided OpalFX as an option if you want to align your sequence inside a SlimShadey session. &nbsp;This option is particularly useful if you export a subsequence to a new tab or delete a sequence from the original alignment and need to realign. Overall, the Opal algorithm has similar performance to MAFTT and Muscle (Wheeler, 2007).</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">What if the shading looks strange or unexpected?</span></p><p class="c3"><span class="c2">SlimShadey is optimized for shading, highlighting, and editing of </span><span class="c13 c2 c18">aligned</span><span class="c1">&nbsp;nucleic acids and protein sequences. &nbsp;You can align the sequences using the algorithm of your choice and import the aligned sequences into SlimShadey as a FASTA file or you can use the included Opal FX to align the sequences in the tab you are working in.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">What if the shading in my RTF file does not match the SlimShadey window?</span></p><p class="c3"><span class="c1">SlimShadey adheres strictly to the rules for the RTF file format. &nbsp;Software such as Microsoft Word may not render all coloring, shading, and characters correctly. To guarantee an accurate representation in Microsoft Word RTF, use grayscale and standard UTF-8 characters. Most colors are supported, though some may be omitted or converted to grayscale by Microsoft Word. We recommend saving to a PNG file for WYSIWYG output.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">When working with sequencing reads the collapse function does not work properly?</span></p><p class="c3"><span class="c1">FASTA files of sequencing reads can contain variable numbers of unknown (N or X) characters at the 5&rsquo; and 3&rsquo; ends. &nbsp;SlimShadey treats sequences with any differences as unique and will not collapse them into a single group. &nbsp;Option 1: The 5&rsquo; and 3&rsquo; ends can be trimmed using the active right-click drag function to delete the columns with unknown residues. &nbsp;Option 2: The active right-click drag function can also be used to select a range of residues and transfer them to a new tab for subsequent analysis. &nbsp;Either approach will allow for the maximum reduction (grouping) of the dataset with reads with unknown residues are eliminated.</span></p><p class="c3 c5"><span class="c1 c15"></span></p><p class="c3"><span class="c19 c2 c18">Can I modify SlimShadey and compile it on my own machine?</span></p><p class="c3"><span class="c2">Yes, however, you will need a functional understanding of JavaFX. &nbsp;All source code for SlimShadey is freely available under GNU General Public License (GPLv3). &nbsp;SlimShadey was developed using </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://netbeans.apache.org/&amp;sa=D&amp;source=editors&amp;ust=1619463442543000&amp;usg=AOvVaw0slM1gIRGtlXTPXY16LNSd">NetBeans</a></span><span class="c1">&nbsp;and the general steps are indicated below:</span></p><ul class="c10 lst-kix_s8ljebqdwq0-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">File &rarr; New Project &rarr; Java with Ant &rarr; Java Application</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Name project SeqVerter and uncheck create main class</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Go to Documents &rarr; &nbsp;NetBeansProjects &rarr; SeqVerter folder</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Replace the src with the latest version (Desktop) </span></li></ul><p class="c3 c5"><span class="c1"></span></p><p class="c3 c5"><span class="c19 c25 c26"></span></p><p class="c3"><span class="c19 c2 c15 c18">Can I write my own shading rules?</span></p><p class="c3"><span class="c1">Yes, though this will require some knowledge of basic Java (no knowledge of JavaFX is required), and a standard procedure for mapping some characteristics of the alignment to RGB colorspace. &nbsp;For the identification of novel motifs and domains, SlimShadey supports shading using ScanProsite regular expressions. If you would like to write your own rules for shading, the rule sets for are in the following classes and data files:</span></p><p class="c3 c5"><span class="c19 c2 c18"></span></p><p class="c3"><span class="c19 c2 c18">Can I modify classes and datafiles?</span></p><p class="c3"><span class="c1">Yes. &nbsp;SlimShadey is provided with all source code. &nbsp;Classes can be modified and recompiled to customize options or add functionality to SlimShadey. &nbsp;Slimshadey was developed using JDK 8 and the NetBeans IDE. &nbsp;If you would like to edit it in the same environment as it was developed then follow the do the following:</span></p><ul class="c10 lst-kix_ahj95ozgae0m-0 start"><li class="c3 c8 li-bullet-0"><span class="c1">Delete NetBeans if it has been previously installed</span></li><li class="c3 c8 li-bullet-0"><span class="c1">Delete all Java versions, JDKs, and updates. </span></li></ul><ul class="c10 lst-kix_ahj95ozgae0m-1 start"><li class="c3 c12 li-bullet-0"><span class="c1">NOTE: If you delete these first then Netbeans uninstaller for Windows 10 will not work. &nbsp;If you deleted them then you need to reinstall them and then delete Netbeans - then delete all Java versions</span></li></ul><ul class="c10 lst-kix_ahj95ozgae0m-0"><li class="c3 c8 li-bullet-0"><span class="c1">Restart</span></li><li class="c3 c8 li-bullet-0"><span class="c2">Install JDK 8u111 with NetBeans 8.2 from Oracle downloadable at this </span><span class="c11 c2"><a class="c6" href="https://www.google.com/url?q=https://www.oracle.com/technetwork/java/javase/downloads/jdk-netbeans-jsp-3413139-esa.html&amp;sa=D&amp;source=editors&amp;ust=1619463442544000&amp;usg=AOvVaw2t-H9_XpB5t4BcOk7--saI">link</a></span></li></ul><ul class="c10 lst-kix_ahj95ozgae0m-1 start"><li class="c3 c12 li-bullet-0"><span class="c2">JDK 8 and NetBeans 8.2 can be obtained separately; however, this installs them together so you don&rsquo;t have to set the paths later and the versions are known to be compatible</span></li></ul><p class="c3 c5"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">SlimShadey on a server</span></p><p class="c3"><span class="c2">SlimShadey will run on most server platforms that support Java. From the terminal check your Java using &ldquo;java -version&rdquo; and output should be in the same general format provided as indicated below. &nbsp;Note that The College of New Jersey ELSA High Performance Computing Cluster (TCNJ HPC) initial check results in Java version &ldquo;1.7.0.261&rdquo;, which does not meet the minimum requirements for SlimShadey. &nbsp;The &ldquo;module add jdk&rdquo; updates the version to &ldquo;1.8.0.102&rdquo;, which meets the minimum requirements. &nbsp;</span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 491.97px; height: 155.30px;"><img alt="" src="images/image2.png" style="width: 491.97px; height: 155.30px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c1">Once the Java version minimum has been met, the SeqVerter.jar file is launched using &ldquo;java -jar &lt;path&gt;&rdquo;, in this case the path is &lsquo;/home/hpc/nayak/SeqVerter.jar&rsquo;. &nbsp;Leaving the terminal console open provides error messages and information on data files used (left). &nbsp;The SlimShadey window running on the TCNJ HPC server is shown on the right.</span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 672.18px; height: 386.30px;"><img alt="" src="images/image4.png" style="width: 672.18px; height: 386.30px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c2">Depending on the server resources and allocations, it may be possible to work with large datasets without collapsing any sequences or experiencing any input lag. &nbsp;The example below shows the sequence alignment of the HOXA1 protein family included in the same data as &ldquo;HOXA1 (large) shaded&rdquo; with the default RASMOL shading. &nbsp;</span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 681.60px; height: 309.33px;"><img alt="" src="images/image3.png" style="width: 681.60px; height: 309.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c1">All aspects of the alignment can be manipulated with no preceptable input lag on the TCNJ HPC, with all rendering performed on CPUs. &nbsp;At the time of this writing the TCNJ HPC has 113 servers, 2,080 CPUs cores, with newer nodes composed of Intel Xeon Gold 6130 (16 core, 32 thread) CPUs with 192GB of RAM/node. &nbsp;</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">If the Java version on your server does not meet the minimum requirements for SlimShadey or you do not have the permissions to update Java, please contact your system administrator. </span></p><p class="c3 c5"><span class="c4 c2"></span></p><hr style="page-break-before:always;display:none;"><p class="c3 c5"><span class="c4 c2"></span></p><p class="c3"><span class="c4 c2">SlimShadey on Linux</span></p><p class="c3"><span class="c1">NOTE: Linux distributions vary so this may have to be modified. &nbsp;This was tested on an HP Chromebook X360 with Debian (Core i3, 8GB RAM, 30GB Linux partition) as Linux (beta) inside Chrome.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">From the terminal, check your Linux kernel version:</span></p><p class="c3"><span class="c7 c2">uname -srm</span></p><p class="c3"><span class="c1">Should return something similar to the following:</span></p><p class="c3"><span class="c7 c2">Linux 5.4.40-04224-g891a6cce2d44 x86_64</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Check your Linux distribution:</span></p><p class="c3"><span class="c7 c2">cat /etc/os-release</span></p><p class="c3"><span class="c1">Should return something similar to the following:</span></p><p class="c3"><span class="c2 c14">PRETTY_NAME=&quot;</span><span class="c2 c9 c17">Debian</span><span class="c7 c2">&nbsp;GNU/Linux 10 (buster)&quot;</span></p><p class="c3"><span class="c7 c2">NAME=&quot;Debian GNU/Linux&quot;</span></p><p class="c3"><span class="c7 c2">VERSION_ID=&quot;10&quot;</span></p><p class="c3"><span class="c7 c2">VERSION=&quot;10 (buster)&quot;</span></p><p class="c3"><span class="c7 c2">VERSION_CODENAME=buster</span></p><p class="c3"><span class="c7 c2">ID=debian</span></p><p class="c3"><span class="c7 c2">HOME_URL=&quot;https://www.debian.org/&quot;</span></p><p class="c3"><span class="c7 c2">SUPPORT_URL=&quot;https://www.debian.org/support&quot;</span></p><p class="c3"><span class="c7 c2">BUG_REPORT_URL=&quot;https://bugs.debian.org/&quot;</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Check current Java version:</span></p><p class="c3"><span class="c7 c2">Java -version</span></p><p class="c3"><span class="c1">If it already says the following then you can skip to &ldquo;Make SlimShadey executable&rdquo;</span></p><p class="c3"><span class="c7 c2">java version &quot;1.8.0_261&quot;</span></p><p class="c3"><span class="c7 c2">Java(TM) SE Runtime Environment (build 1.8.0_261-b12)</span></p><p class="c3"><span class="c14 c2">Java HotSpot(TM) 64-Bit Server VM (build 25.261-b12, mixed mode)</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2">Get Oracle JDK8 from</span><span class="c2 c9">&nbsp;</span><span class="c11 c2 c9"><a class="c6" href="https://www.google.com/url?q=https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html&amp;sa=D&amp;source=editors&amp;ust=1619463442547000&amp;usg=AOvVaw3uCh6Qi5WE_nVQzOWNdi-U">here</a></span><span class="c2 c9">. </span><span class="c2">&nbsp;You may have to create an account (free for non commercial use). &nbsp;This distribution uses Linux x64 Compressed Archive (jdk-</span><span class="c2 c9">8u261</span><span class="c1">-linux-x64.tar.gz). &nbsp;The version in bold will have to be changed in the instructions below if you are using a different version. &nbsp;SlimShadey may work with OpenJDK, however it has not been extensively tested.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Check to see if you already have /usr/lib/jvm</span></p><p class="c3"><span class="c1">If /usr/lib/jvm does not exist then make it with the following:</span></p><p class="c3"><span class="c14 c2">sudo mkdir -p /usr/lib/jvm</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2">Extract jdk-8u261-linux-x64.tar.gz (JDK download) and install in /usr/lib/jvm</span><span class="c4 c2">&nbsp;</span></p><p class="c3"><span class="c1">Use the following if you moved the download to the main directory:</span></p><p class="c3"><span class="c14 c2">sudo tar zxvf jdk-</span><span class="c2 c9 c17">8u261</span><span class="c7 c2">-linux-x64.tar.gz -C /usr/lib/jvm</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Update Java alternative so you can set it as the new default:</span></p><p class="c3"><span class="c14 c2">sudo update-alternatives --install &quot;/usr/bin/java&quot; &quot;java&quot; &quot;/usr/lib/jvm/jdk1.8.0_</span><span class="c2 c9 c17">261</span><span class="c7 c2">/bin/java&quot; 1</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Set the new JDK as the default</span></p><p class="c3"><span class="c14 c2">sudo update-alternatives --set java /usr/lib/jvm/jdk1.8.0_</span><span class="c2 c9 c17">261</span><span class="c7 c2">/bin/java</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Check Java version again:</span></p><p class="c3"><span class="c7 c2">java -version</span></p><p class="c3"><span class="c1">Should output the following:</span></p><p class="c3"><span class="c7 c2">java version &quot;1.8.0_261&quot;</span></p><p class="c3"><span class="c7 c2">Java(TM) SE Runtime Environment (build 1.8.0_261-b12)</span></p><p class="c3"><span class="c7 c2">Java HotSpot(TM) 64-Bit Server VM (build 25.261-b12, mixed mode)</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Make SlimShadey executable:</span></p><p class="c3"><span class="c1">This assumes that you have already unzipped the SlimShadey download and have moved into the main folder where the SeqVerter file is located.</span></p><p class="c3"><span class="c7 c2">Chmod +x SeqVerter.jar</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c19 c2 c18">Run SlimShadey with the following:</span></p><p class="c3"><span class="c14 c2">Java -jar SlimShadey.jar</span></p><p class="c3 c5"><span class="c1"></span></p><hr style="page-break-before:always;display:none;"><p class="c3 c5"><span class="c16 c13 c2 c9"></span></p><p class="c3"><span class="c13 c2 c9">References</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Hulo, N., Sigrist, C.J., Le Saux, V., Langendijk&#8208;Genevaux, P.S., Bordoli, L., Gattiker, A., De Castro, E., Bucher, P. and Bairoch, A., (2004) Recent improvements to the PROSITE database. Nucleic Acids Research, 32(suppl_1), pp.D134-D137</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Jeanmougin F, Thompson JD, Gouy M, Higgins DG, Gibson TJ. Multiple sequence alignment with Clustal X. Trends Biochem Sci. 1998;23(10):403-405. doi:10.1016/s0968-0004(98)01285-7</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Larkin MA, Blackshields G, Brown NP, Chenna R, McGettigan PA, McWilliam H, Valentin F, Wallace IM, Wilm A, Lopez R, Thompson JD, Gibson TJ, Higgins DG.</span></p><p class="c3"><span class="c1">(2007). Clustal W and Clustal X version 2.0. Bioinformatics, 23, 2947-2948.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Lee MH, Schedl T. C. elegans star proteins, GLD-1 and ASD-2, regulate specific RNA targets to control development. Adv Exp Med Biol. 2010;693:106-122. doi:10.1007/978-1-4419-7005-3_8.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Madeira F, Park YM, Lee J, et al. The EMBL-EBI search and sequence analysis tools APIs in 2019. Nucleic Acids Research. 2019 Jul;47(W1):W636-W641. DOI: 10.1093/nar/gkz268.</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Potter SC, Luciani A, Eddy SR, Park Y, Lopez R, Finn RD. HMMER web server: 2018 update. Nucleic Acids Res. 2018;46(W1):W200-W204. doi:10.1093/nar/gky448</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Sayle RA, Milner-White EJ. RASMOL: biomolecular graphics for all. Trends Biochem Sci. 1995;20(9):374. doi:10.1016/s0968-0004(00)89080-5</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Senbanjo LT, Chellaiah MA. CD44: A Multifunctional Cell Surface Adhesion Receptor Is a Regulator of Progression and Metastasis of Cancer Cells. Front Cell Dev Biol. 2017;5:18. Published 2017 Mar 7. doi:10.3389/fcell.2017.00018</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Sigrist CJ, de Castro E, Cerutti L, et al. New and continuing developments at PROSITE. Nucleic Acids Res. 2013;41(Database issue):D344-D347. doi:10.1093/nar/gks1067</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c1">Thompson JD, Higgins DG, Gibson TJ. CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice. Nucleic Acids Res. 1994;22(22):4673-4680. doi:10.1093/nar/22.22.4673</span></p><p class="c3 c5"><span class="c1"></span></p><p class="c3"><span class="c2">Wheeler TJ, Kececioglu JD. Multiple alignment by aligning alignments. Bioinformatics. 2007;23(13):i559-i568. doi:10.1093/bioinformatics/btm226</span></p><div><p class="c5 c20"><span class="c19 c26 c25"></span></p></div><div class="c29"><p class="c21"><a href="#cmnt_ref1" id="cmnt1">[a]</a><span class="c19 c26 c25">Test this</span></p></div></body></html>